{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "id": "X_i8u8SG-3yd"
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import sys\n",
    "import os\n",
    "import re\n",
    "BASE_DIR=\"/private/groups/hprc/qc_hmm_flagger/hprc_intermediate_assembly/assembly_qc\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### This notebook:\n",
    "\n",
    "#### Create HiFi and ONT table for censat batch2 data\n",
    "\n",
    "* There are some samples with new censat annotation from `/private/groups/hprc/qc/batch2/censat/batch2_censat_outputs_done.csv`\n",
    "* Makes diploid censat bed files and add links to the final tables\n",
    "* Takes read paths from tables\n",
    "    * `batch1_jan_12_2025/hmm_flagger/read_tables/hifi_full_reads_table.jan_12_2025.csv`\n",
    "    * `batch1/hmm_flagger/read_tables/ont_reads_table.csv`\n",
    "    * `batch1/hmm_flagger/read_tables/hifi_reads_table.csv`\n",
    "* Makes separate data tables for HiFi and ONT runs (both will contain diploid censat bed files)\n",
    "* Saves the final data tables in `hifi/` and `ont/` subdirectories and they will be used for creating input json files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "diploid_censat_beds  hifi  make_hmm_flagger_data_tables_batch2.ipynb\r\n"
     ]
    }
   ],
   "source": [
    "!ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def mergeHaplotypeRows(censat_table):\n",
    "    # Remove \"_hap1\" or \"_hap2\" from 'sample_id' and create a new column 'sample_name'\n",
    "    censat_table[\"sample_name\"] = censat_table[\"sample_id\"].str.replace(r\"_hap[12]$\", \"\", regex=True)\n",
    "\n",
    "    # make two tables one for hap1 and one for hap2\n",
    "    censat_table_hap1 = censat_table[censat_table['sample_id'].str.endswith('_hap1', na=False)]\n",
    "    censat_table_hap2 = censat_table[censat_table['sample_id'].str.endswith('_hap2', na=False)]\n",
    "\n",
    "    # Merging the DataFrames on 'sample_name'\n",
    "    censat_table_diploid = pd.merge(censat_table_hap1,\n",
    "                                    censat_table_hap2,\n",
    "                                    on='sample_name',\n",
    "                                    suffixes=('_hap1', '_hap2'))\n",
    "    censat_table_diploid = censat_table_diploid[[\"sample_name\", \n",
    "                                                 \"asm_hap1\",\n",
    "                                                 \"asm_hap2\",\n",
    "                                                 \"cenSatAnnotations_hap1\",\n",
    "                                                 \"cenSatAnnotations_hap2\"]]\n",
    "\n",
    "    censat_table_diploid = censat_table_diploid.rename(columns={\"sample_name\":\"sample_id\"})\n",
    "    return censat_table_diploid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def addDiploidCenSatAnnotation(censat_table_diploid, diploid_censat_dir, create_files):\n",
    "    # add a column for saving diploid censat bed files\n",
    "    censat_table_diploid[\"censat_diploid_bed\"] = \"\"\n",
    "\n",
    "    # make a directory for saving diploid censat bed files\n",
    "    !mkdir -p {diploid_censat_dir}\n",
    "\n",
    "    censat_diploid_list = []\n",
    "    additional_annotations_array_list = []\n",
    "    # iterate over rows\n",
    "    for i in range(len(censat_table_diploid)):\n",
    "        sample = censat_table_diploid[\"sample_id\"][i]\n",
    "        censat_bed_hap1 = censat_table_diploid[\"cenSatAnnotations_hap1\"][i]\n",
    "        censat_bed_hap2 = censat_table_diploid[\"cenSatAnnotations_hap2\"][i]\n",
    "\n",
    "        if create_files:\n",
    "            # concat hap1 and hap2 censat bed files into a single bed file\n",
    "            !cat {censat_bed_hap1} {censat_bed_hap2} | bedtools sort -i - > {diploid_censat_dir}/{sample}_dip_hprc_r2_v1.cenSat.bed\n",
    "            !cat {censat_bed_hap1} {censat_bed_hap2} | bedtools sort -i - | grep -i \"rDNA\" | awk '{{print $$1\"\\t\"$$2\"\\t\"$$3}}' > {diploid_censat_dir}/{sample}_dip_hprc_r2_v1.cenSat.rDNA.bed\n",
    "\n",
    "        # add new bed to the table\n",
    "        censat_diploid_list.append(f'{diploid_censat_dir}/{sample}_dip_hprc_r2_v1.cenSat.bed')\n",
    "        # just adding rDNA annotation as an additional annotation\n",
    "        additional_annotations_array_list.append([f'{diploid_censat_dir}/{sample}_dip_hprc_r2_v1.cenSat.rDNA.bed'])\n",
    "\n",
    "    censat_table_diploid[\"censat_diploid_bed\"] = censat_diploid_list\n",
    "    censat_table_diploid[\"additional_annotations_array\"] = additional_annotations_array_list\n",
    "    #censat_table_diploid.head()\n",
    "    return censat_table_diploid"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create HiFi and ONT table for censat batch2 data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of rows: 40 Number of samples: 20.0\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>sample</th>\n",
       "      <th>haplotype</th>\n",
       "      <th>phasing</th>\n",
       "      <th>assembly_method</th>\n",
       "      <th>assembly_method_version</th>\n",
       "      <th>assembly_date</th>\n",
       "      <th>genbank_accession</th>\n",
       "      <th>assembly_md5</th>\n",
       "      <th>assembly_fai</th>\n",
       "      <th>...</th>\n",
       "      <th>rmRmskBigBed</th>\n",
       "      <th>repeatMaskerTarGZ</th>\n",
       "      <th>rmOutFile</th>\n",
       "      <th>rmFinalMaskedFasta</th>\n",
       "      <th>rmRmskAlignBed</th>\n",
       "      <th>rmRmskBed</th>\n",
       "      <th>rmBed</th>\n",
       "      <th>centromeres</th>\n",
       "      <th>cenSatStrand</th>\n",
       "      <th>cenSatAnnotations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG01975_hap1</td>\n",
       "      <td>HG01975</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>hifiasm</td>\n",
       "      <td>0.19.7</td>\n",
       "      <td>2024-12</td>\n",
       "      <td>GCA_046000135.1</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG01975_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG01975_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG01975_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG01975_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG01975_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG01975_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG01975_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG01975_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG01975_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG02602_hap1</td>\n",
       "      <td>HG02602</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>hifiasm</td>\n",
       "      <td>0.19.9</td>\n",
       "      <td>2024-12</td>\n",
       "      <td>GCA_046000095.1</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02602_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02602_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02602_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02602_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02602_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02602_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02602_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02602_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02602_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG04187_hap1</td>\n",
       "      <td>HG04187</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>hifiasm</td>\n",
       "      <td>0.19.9</td>\n",
       "      <td>2024-12</td>\n",
       "      <td>GCA_046000115.1</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG04187_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG04187_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG04187_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG04187_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG04187_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG04187_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG04187_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG04187_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG04187_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NA18879_hap1</td>\n",
       "      <td>NA18879</td>\n",
       "      <td>1</td>\n",
       "      <td>hic</td>\n",
       "      <td>hifiasm</td>\n",
       "      <td>0.19.9</td>\n",
       "      <td>2024-12</td>\n",
       "      <td>GCA_046000005.1</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA18879_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA18879_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA18879_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA18879_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA18879_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA18879_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA18879_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA18879_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA18879_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NA20752_hap1</td>\n",
       "      <td>NA20752</td>\n",
       "      <td>1</td>\n",
       "      <td>hic</td>\n",
       "      <td>hifiasm</td>\n",
       "      <td>0.19.9</td>\n",
       "      <td>2024-12</td>\n",
       "      <td>GCA_046000145.1</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA20752_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA20752_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA20752_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA20752_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA20752_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA20752_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA20752_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA20752_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA20752_...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      sample_id   sample  haplotype phasing assembly_method  \\\n",
       "0  HG01975_hap1  HG01975          1    trio         hifiasm   \n",
       "1  HG02602_hap1  HG02602          1    trio         hifiasm   \n",
       "2  HG04187_hap1  HG04187          1    trio         hifiasm   \n",
       "3  NA18879_hap1  NA18879          1     hic         hifiasm   \n",
       "4  NA20752_hap1  NA20752          1     hic         hifiasm   \n",
       "\n",
       "  assembly_method_version assembly_date genbank_accession  \\\n",
       "0                  0.19.7       2024-12   GCA_046000135.1   \n",
       "1                  0.19.9       2024-12   GCA_046000095.1   \n",
       "2                  0.19.9       2024-12   GCA_046000115.1   \n",
       "3                  0.19.9       2024-12   GCA_046000005.1   \n",
       "4                  0.19.9       2024-12   GCA_046000145.1   \n",
       "\n",
       "                                        assembly_md5  \\\n",
       "0  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "1  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "2  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "3  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "4  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "\n",
       "                                        assembly_fai  \\\n",
       "0  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "1  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "2  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "3  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "4  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "\n",
       "                         ...                         rmRmskBigBed  \\\n",
       "0                        ...                                  NaN   \n",
       "1                        ...                                  NaN   \n",
       "2                        ...                                  NaN   \n",
       "3                        ...                                  NaN   \n",
       "4                        ...                                  NaN   \n",
       "\n",
       "                                   repeatMaskerTarGZ  \\\n",
       "0  /private/groups/hprc/qc/batch2/censat/HG01975_...   \n",
       "1  /private/groups/hprc/qc/batch2/censat/HG02602_...   \n",
       "2  /private/groups/hprc/qc/batch2/censat/HG04187_...   \n",
       "3  /private/groups/hprc/qc/batch2/censat/NA18879_...   \n",
       "4  /private/groups/hprc/qc/batch2/censat/NA20752_...   \n",
       "\n",
       "                                           rmOutFile  \\\n",
       "0  /private/groups/hprc/qc/batch2/censat/HG01975_...   \n",
       "1  /private/groups/hprc/qc/batch2/censat/HG02602_...   \n",
       "2  /private/groups/hprc/qc/batch2/censat/HG04187_...   \n",
       "3  /private/groups/hprc/qc/batch2/censat/NA18879_...   \n",
       "4  /private/groups/hprc/qc/batch2/censat/NA20752_...   \n",
       "\n",
       "                                  rmFinalMaskedFasta  \\\n",
       "0  /private/groups/hprc/qc/batch2/censat/HG01975_...   \n",
       "1  /private/groups/hprc/qc/batch2/censat/HG02602_...   \n",
       "2  /private/groups/hprc/qc/batch2/censat/HG04187_...   \n",
       "3  /private/groups/hprc/qc/batch2/censat/NA18879_...   \n",
       "4  /private/groups/hprc/qc/batch2/censat/NA20752_...   \n",
       "\n",
       "                                      rmRmskAlignBed  \\\n",
       "0  /private/groups/hprc/qc/batch2/censat/HG01975_...   \n",
       "1  /private/groups/hprc/qc/batch2/censat/HG02602_...   \n",
       "2  /private/groups/hprc/qc/batch2/censat/HG04187_...   \n",
       "3  /private/groups/hprc/qc/batch2/censat/NA18879_...   \n",
       "4  /private/groups/hprc/qc/batch2/censat/NA20752_...   \n",
       "\n",
       "                                           rmRmskBed  \\\n",
       "0  /private/groups/hprc/qc/batch2/censat/HG01975_...   \n",
       "1  /private/groups/hprc/qc/batch2/censat/HG02602_...   \n",
       "2  /private/groups/hprc/qc/batch2/censat/HG04187_...   \n",
       "3  /private/groups/hprc/qc/batch2/censat/NA18879_...   \n",
       "4  /private/groups/hprc/qc/batch2/censat/NA20752_...   \n",
       "\n",
       "                                               rmBed  \\\n",
       "0  /private/groups/hprc/qc/batch2/censat/HG01975_...   \n",
       "1  /private/groups/hprc/qc/batch2/censat/HG02602_...   \n",
       "2  /private/groups/hprc/qc/batch2/censat/HG04187_...   \n",
       "3  /private/groups/hprc/qc/batch2/censat/NA18879_...   \n",
       "4  /private/groups/hprc/qc/batch2/censat/NA20752_...   \n",
       "\n",
       "                                         centromeres  \\\n",
       "0  /private/groups/hprc/qc/batch2/censat/HG01975_...   \n",
       "1  /private/groups/hprc/qc/batch2/censat/HG02602_...   \n",
       "2  /private/groups/hprc/qc/batch2/censat/HG04187_...   \n",
       "3  /private/groups/hprc/qc/batch2/censat/NA18879_...   \n",
       "4  /private/groups/hprc/qc/batch2/censat/NA20752_...   \n",
       "\n",
       "                                        cenSatStrand  \\\n",
       "0  /private/groups/hprc/qc/batch2/censat/HG01975_...   \n",
       "1  /private/groups/hprc/qc/batch2/censat/HG02602_...   \n",
       "2  /private/groups/hprc/qc/batch2/censat/HG04187_...   \n",
       "3  /private/groups/hprc/qc/batch2/censat/NA18879_...   \n",
       "4  /private/groups/hprc/qc/batch2/censat/NA20752_...   \n",
       "\n",
       "                                   cenSatAnnotations  \n",
       "0  /private/groups/hprc/qc/batch2/censat/HG01975_...  \n",
       "1  /private/groups/hprc/qc/batch2/censat/HG02602_...  \n",
       "2  /private/groups/hprc/qc/batch2/censat/HG04187_...  \n",
       "3  /private/groups/hprc/qc/batch2/censat/NA18879_...  \n",
       "4  /private/groups/hprc/qc/batch2/censat/NA20752_...  \n",
       "\n",
       "[5 rows x 28 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "censat_table_batch2 = pd.read_csv('/private/groups/hprc/qc/batch2/censat/batch2_censat_outputs_done.csv')\n",
    "print(\"Number of rows:\", len(censat_table_batch2), \"Number of samples:\", len(censat_table_batch2) / 2)\n",
    "censat_table_batch2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# since in batch2 we have \"assembly\" column instead of \"asm\"\n",
    "# let's rename it so that mergeHaplotypeRows() work properly\n",
    "censat_table_batch2 = censat_table_batch2.rename(columns={\"assembly\": \"asm\"})\n",
    "\n",
    "# merge rows for two haplotypes into one row with columns having _hap1 and _hap2 suffixes\n",
    "censat_table_diploid_batch2 = mergeHaplotypeRows(censat_table_batch2)\n",
    "\n",
    "# a directory for saving/getting diploid censat bed files\n",
    "diploid_censat_dir_batch2 = f'{BASE_DIR}/batch2/hmm_flagger/diploid_censat_beds'\n",
    "# I set create_files to False since I ran it once with True before\n",
    "censat_table_diploid_batch2 = addDiploidCenSatAnnotation(censat_table_diploid = censat_table_diploid_batch2, \n",
    "                                                         diploid_censat_dir = diploid_censat_dir_batch2, \n",
    "                                                         create_files = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HG01975#1#CM099804.1\t35330208\t35333289\r\n",
      "HG01975#1#CM099806.1\t30102163\t30105211\r\n",
      "HG01975#1#CM099806.1\t31025828\t31028888\r\n",
      "HG01975#1#JBJUVS010000002.1\t2147\t45912\r\n",
      "HG01975#1#JBJUVS010000002.1\t54577\t330802\r\n",
      "HG01975#1#JBJUVS010000002.1\t2864167\t2867245\r\n",
      "HG01975#1#JBJUVS010000004.1\t19145\t203683\r\n",
      "HG01975#1#JBJUVS010000004.1\t6113313\t6116368\r\n",
      "HG01975#1#JBJUVS010000005.1\t24626\t258972\r\n",
      "HG01975#1#JBJUVS010000005.1\t2800717\t2803795\r\n"
     ]
    }
   ],
   "source": [
    "!cat {BASE_DIR}/batch2/hmm_flagger/diploid_censat_beds/HG01975_dip_hprc_r2_v1.cenSat.rDNA.bed | head"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### ONT table for batch2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# parse ont reads table\n",
    "ont_reads_table = pd.read_csv(f'{BASE_DIR}/batch1/hmm_flagger/read_tables/ont_reads_table.csv')\n",
    "\n",
    "# merge with censat batch2 table\n",
    "merged_ont_data_table_batch2 = pd.merge(censat_table_diploid_batch2, ont_reads_table, on='sample_id',  how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of rows for final ONT table: 12\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>asm_hap1</th>\n",
       "      <th>asm_hap2</th>\n",
       "      <th>cenSatAnnotations_hap1</th>\n",
       "      <th>cenSatAnnotations_hap2</th>\n",
       "      <th>censat_diploid_bed</th>\n",
       "      <th>additional_annotations_array</th>\n",
       "      <th>read_files_downsampled</th>\n",
       "      <th>number_of_read_files_downsampled</th>\n",
       "      <th>total_coverage_downsampled</th>\n",
       "      <th>...</th>\n",
       "      <th>kmer_size</th>\n",
       "      <th>read_files</th>\n",
       "      <th>number_of_read_files</th>\n",
       "      <th>total_coverage</th>\n",
       "      <th>coverage</th>\n",
       "      <th>number_of_cores_per_task</th>\n",
       "      <th>sequencing_chemistry</th>\n",
       "      <th>hmm_flagger_window_size</th>\n",
       "      <th>hmm_flagger_alpha_tsv</th>\n",
       "      <th>suffix_mapping</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG01975</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG01975_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG01975_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG01975/...</td>\n",
       "      <td>3</td>\n",
       "      <td>78.43</td>\n",
       "      <td>...</td>\n",
       "      <td>15</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG01975/...</td>\n",
       "      <td>3</td>\n",
       "      <td>78.43</td>\n",
       "      <td>[33.07, 22.43, 22.93]</td>\n",
       "      <td>21</td>\n",
       "      <td>R941</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "      <td>R941_minimap2_2.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG02602</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02602_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02602_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02602/...</td>\n",
       "      <td>2</td>\n",
       "      <td>61.43</td>\n",
       "      <td>...</td>\n",
       "      <td>15</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02602/...</td>\n",
       "      <td>3</td>\n",
       "      <td>83.09</td>\n",
       "      <td>[32.31, 29.12, 21.66]</td>\n",
       "      <td>21</td>\n",
       "      <td>R941</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "      <td>R941_minimap2_2.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG04187</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG04187_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG04187_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG04187/...</td>\n",
       "      <td>4</td>\n",
       "      <td>47.10</td>\n",
       "      <td>...</td>\n",
       "      <td>15</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG04187/...</td>\n",
       "      <td>4</td>\n",
       "      <td>47.10</td>\n",
       "      <td>[7.78, 20.02, 18.44, 0.86]</td>\n",
       "      <td>16</td>\n",
       "      <td>R941</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "      <td>R941_minimap2_2.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NA18879</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA18879_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA18879_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/NA18879/...</td>\n",
       "      <td>3</td>\n",
       "      <td>54.67</td>\n",
       "      <td>...</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/NA18879/...</td>\n",
       "      <td>3</td>\n",
       "      <td>54.67</td>\n",
       "      <td>[18.49, 17.84, 18.34]</td>\n",
       "      <td>21</td>\n",
       "      <td>R1041</td>\n",
       "      <td>8000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "      <td>R1041_minimap2_2.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG02738</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02738_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02738_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02738/...</td>\n",
       "      <td>3</td>\n",
       "      <td>56.66</td>\n",
       "      <td>...</td>\n",
       "      <td>15</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02738/...</td>\n",
       "      <td>3</td>\n",
       "      <td>56.66</td>\n",
       "      <td>[6.17, 21.49, 29.0]</td>\n",
       "      <td>21</td>\n",
       "      <td>R941</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "      <td>R941_minimap2_2.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HG00706</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG00706_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG00706_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG00706/...</td>\n",
       "      <td>3</td>\n",
       "      <td>72.21</td>\n",
       "      <td>...</td>\n",
       "      <td>15</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG00706/...</td>\n",
       "      <td>3</td>\n",
       "      <td>72.21</td>\n",
       "      <td>[22.45, 27.23, 22.53]</td>\n",
       "      <td>21</td>\n",
       "      <td>R941</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "      <td>R941_minimap2_2.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HG03816</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG03816_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG03816_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG03816/...</td>\n",
       "      <td>3</td>\n",
       "      <td>79.52</td>\n",
       "      <td>...</td>\n",
       "      <td>15</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG03816/...</td>\n",
       "      <td>3</td>\n",
       "      <td>79.52</td>\n",
       "      <td>[28.94, 20.57, 30.01]</td>\n",
       "      <td>21</td>\n",
       "      <td>R941</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "      <td>R941_minimap2_2.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>HG02392</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02392_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02392_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02392/...</td>\n",
       "      <td>3</td>\n",
       "      <td>76.80</td>\n",
       "      <td>...</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02392/...</td>\n",
       "      <td>3</td>\n",
       "      <td>76.80</td>\n",
       "      <td>[24.26, 22.84, 29.7]</td>\n",
       "      <td>21</td>\n",
       "      <td>R1041</td>\n",
       "      <td>8000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "      <td>R1041_minimap2_2.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>HG02514</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02514_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02514_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02514/...</td>\n",
       "      <td>3</td>\n",
       "      <td>64.07</td>\n",
       "      <td>...</td>\n",
       "      <td>15</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02514/...</td>\n",
       "      <td>3</td>\n",
       "      <td>64.07</td>\n",
       "      <td>[21.49, 22.59, 19.99]</td>\n",
       "      <td>21</td>\n",
       "      <td>R941</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "      <td>R941_minimap2_2.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HG02841</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02841_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02841_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02841/...</td>\n",
       "      <td>3</td>\n",
       "      <td>59.53</td>\n",
       "      <td>...</td>\n",
       "      <td>15</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02841/...</td>\n",
       "      <td>3</td>\n",
       "      <td>59.53</td>\n",
       "      <td>[19.14, 13.88, 26.51]</td>\n",
       "      <td>21</td>\n",
       "      <td>R941</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "      <td>R941_minimap2_2.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>HG02984</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02984_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02984_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02984/...</td>\n",
       "      <td>3</td>\n",
       "      <td>64.70</td>\n",
       "      <td>...</td>\n",
       "      <td>15</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02984/...</td>\n",
       "      <td>4</td>\n",
       "      <td>71.35</td>\n",
       "      <td>[30.86, 23.24, 10.6, 6.65]</td>\n",
       "      <td>16</td>\n",
       "      <td>R941</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "      <td>R941_minimap2_2.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>HG03050</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG03050_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG03050_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG03050/...</td>\n",
       "      <td>3</td>\n",
       "      <td>60.55</td>\n",
       "      <td>...</td>\n",
       "      <td>15</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG03050/...</td>\n",
       "      <td>3</td>\n",
       "      <td>60.55</td>\n",
       "      <td>[23.24, 15.72, 21.59]</td>\n",
       "      <td>21</td>\n",
       "      <td>R941</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "      <td>R941_minimap2_2.28</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   sample_id                                           asm_hap1  \\\n",
       "0    HG01975  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "1    HG02602  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "2    HG04187  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "3    NA18879  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "4    HG02738  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "5    HG00706  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "6    HG03816  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "7    HG02392  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "8    HG02514  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "9    HG02841  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "10   HG02984  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "11   HG03050  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "\n",
       "                                             asm_hap2  \\\n",
       "0   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "1   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "2   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "3   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "4   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "5   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "6   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "7   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "8   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "9   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "10  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "11  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "\n",
       "                               cenSatAnnotations_hap1  \\\n",
       "0   /private/groups/hprc/qc/batch2/censat/HG01975_...   \n",
       "1   /private/groups/hprc/qc/batch2/censat/HG02602_...   \n",
       "2   /private/groups/hprc/qc/batch2/censat/HG04187_...   \n",
       "3   /private/groups/hprc/qc/batch2/censat/NA18879_...   \n",
       "4   /private/groups/hprc/qc/batch2/censat/HG02738_...   \n",
       "5   /private/groups/hprc/qc/batch2/censat/HG00706_...   \n",
       "6   /private/groups/hprc/qc/batch2/censat/HG03816_...   \n",
       "7   /private/groups/hprc/qc/batch2/censat/HG02392_...   \n",
       "8   /private/groups/hprc/qc/batch2/censat/HG02514_...   \n",
       "9   /private/groups/hprc/qc/batch2/censat/HG02841_...   \n",
       "10  /private/groups/hprc/qc/batch2/censat/HG02984_...   \n",
       "11  /private/groups/hprc/qc/batch2/censat/HG03050_...   \n",
       "\n",
       "                               cenSatAnnotations_hap2  \\\n",
       "0   /private/groups/hprc/qc/batch2/censat/HG01975_...   \n",
       "1   /private/groups/hprc/qc/batch2/censat/HG02602_...   \n",
       "2   /private/groups/hprc/qc/batch2/censat/HG04187_...   \n",
       "3   /private/groups/hprc/qc/batch2/censat/NA18879_...   \n",
       "4   /private/groups/hprc/qc/batch2/censat/HG02738_...   \n",
       "5   /private/groups/hprc/qc/batch2/censat/HG00706_...   \n",
       "6   /private/groups/hprc/qc/batch2/censat/HG03816_...   \n",
       "7   /private/groups/hprc/qc/batch2/censat/HG02392_...   \n",
       "8   /private/groups/hprc/qc/batch2/censat/HG02514_...   \n",
       "9   /private/groups/hprc/qc/batch2/censat/HG02841_...   \n",
       "10  /private/groups/hprc/qc/batch2/censat/HG02984_...   \n",
       "11  /private/groups/hprc/qc/batch2/censat/HG03050_...   \n",
       "\n",
       "                                   censat_diploid_bed  \\\n",
       "0   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "1   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "2   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "3   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "4   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "5   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "6   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "7   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "8   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "9   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "10  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "11  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "\n",
       "                         additional_annotations_array  \\\n",
       "0   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "1   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "2   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "3   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "4   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "5   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "6   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "7   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "8   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "9   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "10  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "11  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "\n",
       "                               read_files_downsampled  \\\n",
       "0   ['s3://human-pangenomics/working/HPRC/HG01975/...   \n",
       "1   ['s3://human-pangenomics/working/HPRC/HG02602/...   \n",
       "2   ['s3://human-pangenomics/working/HPRC/HG04187/...   \n",
       "3   ['s3://human-pangenomics/working/HPRC/NA18879/...   \n",
       "4   ['s3://human-pangenomics/working/HPRC/HG02738/...   \n",
       "5   ['s3://human-pangenomics/working/HPRC/HG00706/...   \n",
       "6   ['s3://human-pangenomics/working/HPRC/HG03816/...   \n",
       "7   ['s3://human-pangenomics/working/HPRC/HG02392/...   \n",
       "8   ['s3://human-pangenomics/working/HPRC/HG02514/...   \n",
       "9   ['s3://human-pangenomics/working/HPRC/HG02841/...   \n",
       "10  ['s3://human-pangenomics/working/HPRC/HG02984/...   \n",
       "11  ['s3://human-pangenomics/working/HPRC/HG03050/...   \n",
       "\n",
       "    number_of_read_files_downsampled  total_coverage_downsampled  \\\n",
       "0                                  3                       78.43   \n",
       "1                                  2                       61.43   \n",
       "2                                  4                       47.10   \n",
       "3                                  3                       54.67   \n",
       "4                                  3                       56.66   \n",
       "5                                  3                       72.21   \n",
       "6                                  3                       79.52   \n",
       "7                                  3                       76.80   \n",
       "8                                  3                       64.07   \n",
       "9                                  3                       59.53   \n",
       "10                                 3                       64.70   \n",
       "11                                 3                       60.55   \n",
       "\n",
       "           ...           kmer_size  \\\n",
       "0          ...                  15   \n",
       "1          ...                  15   \n",
       "2          ...                  15   \n",
       "3          ...                  25   \n",
       "4          ...                  15   \n",
       "5          ...                  15   \n",
       "6          ...                  15   \n",
       "7          ...                  25   \n",
       "8          ...                  15   \n",
       "9          ...                  15   \n",
       "10         ...                  15   \n",
       "11         ...                  15   \n",
       "\n",
       "                                           read_files  number_of_read_files  \\\n",
       "0   ['s3://human-pangenomics/working/HPRC/HG01975/...                     3   \n",
       "1   ['s3://human-pangenomics/working/HPRC/HG02602/...                     3   \n",
       "2   ['s3://human-pangenomics/working/HPRC/HG04187/...                     4   \n",
       "3   ['s3://human-pangenomics/working/HPRC/NA18879/...                     3   \n",
       "4   ['s3://human-pangenomics/working/HPRC/HG02738/...                     3   \n",
       "5   ['s3://human-pangenomics/working/HPRC/HG00706/...                     3   \n",
       "6   ['s3://human-pangenomics/working/HPRC/HG03816/...                     3   \n",
       "7   ['s3://human-pangenomics/working/HPRC/HG02392/...                     3   \n",
       "8   ['s3://human-pangenomics/working/HPRC/HG02514/...                     3   \n",
       "9   ['s3://human-pangenomics/working/HPRC/HG02841/...                     3   \n",
       "10  ['s3://human-pangenomics/working/HPRC/HG02984/...                     4   \n",
       "11  ['s3://human-pangenomics/working/HPRC/HG03050/...                     3   \n",
       "\n",
       "   total_coverage                    coverage  number_of_cores_per_task  \\\n",
       "0           78.43       [33.07, 22.43, 22.93]                        21   \n",
       "1           83.09       [32.31, 29.12, 21.66]                        21   \n",
       "2           47.10  [7.78, 20.02, 18.44, 0.86]                        16   \n",
       "3           54.67       [18.49, 17.84, 18.34]                        21   \n",
       "4           56.66         [6.17, 21.49, 29.0]                        21   \n",
       "5           72.21       [22.45, 27.23, 22.53]                        21   \n",
       "6           79.52       [28.94, 20.57, 30.01]                        21   \n",
       "7           76.80        [24.26, 22.84, 29.7]                        21   \n",
       "8           64.07       [21.49, 22.59, 19.99]                        21   \n",
       "9           59.53       [19.14, 13.88, 26.51]                        21   \n",
       "10          71.35  [30.86, 23.24, 10.6, 6.65]                        16   \n",
       "11          60.55       [23.24, 15.72, 21.59]                        21   \n",
       "\n",
       "   sequencing_chemistry  hmm_flagger_window_size  \\\n",
       "0                  R941                    16000   \n",
       "1                  R941                    16000   \n",
       "2                  R941                    16000   \n",
       "3                 R1041                     8000   \n",
       "4                  R941                    16000   \n",
       "5                  R941                    16000   \n",
       "6                  R941                    16000   \n",
       "7                 R1041                     8000   \n",
       "8                  R941                    16000   \n",
       "9                  R941                    16000   \n",
       "10                 R941                    16000   \n",
       "11                 R941                    16000   \n",
       "\n",
       "                                hmm_flagger_alpha_tsv       suffix_mapping  \n",
       "0   https://raw.githubusercontent.com/mobinasri/fl...   R941_minimap2_2.28  \n",
       "1   https://raw.githubusercontent.com/mobinasri/fl...   R941_minimap2_2.28  \n",
       "2   https://raw.githubusercontent.com/mobinasri/fl...   R941_minimap2_2.28  \n",
       "3   https://raw.githubusercontent.com/mobinasri/fl...  R1041_minimap2_2.28  \n",
       "4   https://raw.githubusercontent.com/mobinasri/fl...   R941_minimap2_2.28  \n",
       "5   https://raw.githubusercontent.com/mobinasri/fl...   R941_minimap2_2.28  \n",
       "6   https://raw.githubusercontent.com/mobinasri/fl...   R941_minimap2_2.28  \n",
       "7   https://raw.githubusercontent.com/mobinasri/fl...  R1041_minimap2_2.28  \n",
       "8   https://raw.githubusercontent.com/mobinasri/fl...   R941_minimap2_2.28  \n",
       "9   https://raw.githubusercontent.com/mobinasri/fl...   R941_minimap2_2.28  \n",
       "10  https://raw.githubusercontent.com/mobinasri/fl...   R941_minimap2_2.28  \n",
       "11  https://raw.githubusercontent.com/mobinasri/fl...   R941_minimap2_2.28  \n",
       "\n",
       "[12 rows x 22 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(f\"Number of rows for final ONT table: {len(merged_ont_data_table_batch2)}\")\n",
    "merged_ont_data_table_batch2[\"suffix_mapping\"] = merged_ont_data_table_batch2[\"sequencing_chemistry\"] + \"_minimap2_2.28\"\n",
    "merged_ont_data_table_batch2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "These 8 samples don't have ONT reads for censat batch 2: \n",
      "\n",
      "NA20752\n",
      "NA19159\n",
      "HG005\n",
      "HG02145\n",
      "NA19240\n",
      "HG00733\n",
      "HG01243\n",
      "NA20799\n"
     ]
    }
   ],
   "source": [
    "missing_samples = set(censat_table_diploid_batch2['sample_id']).difference(merged_ont_data_table_batch2['sample_id'])\n",
    "print(f\"These {len(missing_samples)} samples don't have ONT reads for censat batch 2: \\n\")\n",
    "print(\"\\n\".join(missing_samples))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### HiFi table for batch2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# parse hifi reads table (jan 12 2025 version containing both Revio and DeepConsensus)\n",
    "hifi_full_reads_table = pd.read_csv(f'{BASE_DIR}/batch1_jan_12_2025/hmm_flagger/read_tables/hifi_full_reads_table.jan_12_2025.csv')\n",
    "\n",
    "# parse hifi Deepconsensus reads table\n",
    "# this table contains only DeepConsensus reads and I used it for batch1 of HMM-Flagger runs\n",
    "# there are some samples with higher coverage in hifi_additional_reads_table\n",
    "# so I prioritize hifi_full_reads_table\n",
    "hifi_dc_reads_table = pd.read_csv(f'{BASE_DIR}/batch1/hmm_flagger/read_tables/hifi_reads_table.csv')\n",
    "\n",
    "# merge with censat batch2 table\n",
    "hifi_full_data_table_batch2 = pd.merge(censat_table_diploid_batch2, hifi_full_reads_table, on='sample_id',  how='inner')\n",
    "\n",
    "# if there are some common samples between \"hifi_additional_reads_table\" and \"hifi_reads_table\"\n",
    "# here I prioritize \"hifi_additional_reads_table\" over \"hifi_reads_table\"\n",
    "# since the coverage is generally higher in the former one\n",
    "samples_with_hifi_full_batch2 = hifi_full_data_table_batch2['sample_id']\n",
    "\n",
    "hifi_dc_data_table_batch2 = pd.merge(censat_table_diploid_batch2, hifi_dc_reads_table, on='sample_id',  how='inner')\n",
    "# remove common samples\n",
    "hifi_dc_data_table_batch2 = hifi_dc_data_table_batch2[~hifi_dc_data_table_batch2['sample_id'].isin(samples_with_hifi_full_batch2)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of rows in hifi full reads table (jan 12 2025 version) with censat batch2: 15\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>asm_hap1</th>\n",
       "      <th>asm_hap2</th>\n",
       "      <th>cenSatAnnotations_hap1</th>\n",
       "      <th>cenSatAnnotations_hap2</th>\n",
       "      <th>censat_diploid_bed</th>\n",
       "      <th>additional_annotations_array</th>\n",
       "      <th>read_files_downsampled</th>\n",
       "      <th>number_of_read_files_downsampled</th>\n",
       "      <th>total_coverage_downsampled</th>\n",
       "      <th>...</th>\n",
       "      <th>mapper_preset</th>\n",
       "      <th>kmer_size</th>\n",
       "      <th>read_files</th>\n",
       "      <th>number_of_read_files</th>\n",
       "      <th>total_coverage</th>\n",
       "      <th>coverage</th>\n",
       "      <th>number_of_cores_per_task</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>hmm_flagger_window_size</th>\n",
       "      <th>hmm_flagger_alpha_tsv</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG01975</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG01975_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG01975_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG01975/...</td>\n",
       "      <td>6</td>\n",
       "      <td>58.8</td>\n",
       "      <td>...</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG01975/...</td>\n",
       "      <td>6</td>\n",
       "      <td>58.8</td>\n",
       "      <td>[12.1, 11.7, 1.9, 9.4, 11.7, 12.0]</td>\n",
       "      <td>10</td>\n",
       "      <td>['Sequel II', 'Sequel II', 'Sequel II', 'Seque...</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG02602</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02602_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02602_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02602/...</td>\n",
       "      <td>7</td>\n",
       "      <td>59.9</td>\n",
       "      <td>...</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02602/...</td>\n",
       "      <td>7</td>\n",
       "      <td>59.9</td>\n",
       "      <td>[6.4, 8.1, 6.4, 11.7, 8.8, 9.5, 9.0]</td>\n",
       "      <td>9</td>\n",
       "      <td>['Sequel II', 'Sequel II', 'Sequel II', 'Seque...</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG04187</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG04187_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG04187_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG04187/...</td>\n",
       "      <td>1</td>\n",
       "      <td>28.4</td>\n",
       "      <td>...</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG04187/...</td>\n",
       "      <td>1</td>\n",
       "      <td>28.4</td>\n",
       "      <td>[28.4]</td>\n",
       "      <td>64</td>\n",
       "      <td>['Revio']</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NA18879</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA18879_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA18879_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/NA18879/...</td>\n",
       "      <td>5</td>\n",
       "      <td>63.8</td>\n",
       "      <td>...</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/NA18879/...</td>\n",
       "      <td>5</td>\n",
       "      <td>63.8</td>\n",
       "      <td>[10.9, 11.6, 11.4, 11.0, 18.9]</td>\n",
       "      <td>12</td>\n",
       "      <td>['Revio', 'Revio', 'Revio', 'Revio', 'Revio']</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NA20752</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA20752_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA20752_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/NA20752/...</td>\n",
       "      <td>4</td>\n",
       "      <td>65.5</td>\n",
       "      <td>...</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/NA20752/...</td>\n",
       "      <td>4</td>\n",
       "      <td>65.5</td>\n",
       "      <td>[15.5, 15.1, 13.6, 21.3]</td>\n",
       "      <td>16</td>\n",
       "      <td>['Sequel II', 'Sequel II', 'Sequel II', 'Revio']</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  sample_id                                           asm_hap1  \\\n",
       "0   HG01975  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "1   HG02602  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "2   HG04187  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "3   NA18879  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "4   NA20752  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "\n",
       "                                            asm_hap2  \\\n",
       "0  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "1  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "2  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "3  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "4  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "\n",
       "                              cenSatAnnotations_hap1  \\\n",
       "0  /private/groups/hprc/qc/batch2/censat/HG01975_...   \n",
       "1  /private/groups/hprc/qc/batch2/censat/HG02602_...   \n",
       "2  /private/groups/hprc/qc/batch2/censat/HG04187_...   \n",
       "3  /private/groups/hprc/qc/batch2/censat/NA18879_...   \n",
       "4  /private/groups/hprc/qc/batch2/censat/NA20752_...   \n",
       "\n",
       "                              cenSatAnnotations_hap2  \\\n",
       "0  /private/groups/hprc/qc/batch2/censat/HG01975_...   \n",
       "1  /private/groups/hprc/qc/batch2/censat/HG02602_...   \n",
       "2  /private/groups/hprc/qc/batch2/censat/HG04187_...   \n",
       "3  /private/groups/hprc/qc/batch2/censat/NA18879_...   \n",
       "4  /private/groups/hprc/qc/batch2/censat/NA20752_...   \n",
       "\n",
       "                                  censat_diploid_bed  \\\n",
       "0  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "1  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "2  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "3  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "4  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "\n",
       "                        additional_annotations_array  \\\n",
       "0  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "1  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "2  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "3  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "4  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "\n",
       "                              read_files_downsampled  \\\n",
       "0  ['s3://human-pangenomics/working/HPRC/HG01975/...   \n",
       "1  ['s3://human-pangenomics/working/HPRC/HG02602/...   \n",
       "2  ['s3://human-pangenomics/working/HPRC/HG04187/...   \n",
       "3  ['s3://human-pangenomics/working/HPRC/NA18879/...   \n",
       "4  ['s3://human-pangenomics/working/HPRC/NA20752/...   \n",
       "\n",
       "   number_of_read_files_downsampled  total_coverage_downsampled  \\\n",
       "0                                 6                        58.8   \n",
       "1                                 7                        59.9   \n",
       "2                                 1                        28.4   \n",
       "3                                 5                        63.8   \n",
       "4                                 4                        65.5   \n",
       "\n",
       "                         ...                          mapper_preset kmer_size  \\\n",
       "0                        ...                                lr:hqae        25   \n",
       "1                        ...                                lr:hqae        25   \n",
       "2                        ...                                lr:hqae        25   \n",
       "3                        ...                                lr:hqae        25   \n",
       "4                        ...                                lr:hqae        25   \n",
       "\n",
       "                                          read_files number_of_read_files  \\\n",
       "0  ['s3://human-pangenomics/working/HPRC/HG01975/...                    6   \n",
       "1  ['s3://human-pangenomics/working/HPRC/HG02602/...                    7   \n",
       "2  ['s3://human-pangenomics/working/HPRC/HG04187/...                    1   \n",
       "3  ['s3://human-pangenomics/working/HPRC/NA18879/...                    5   \n",
       "4  ['s3://human-pangenomics/working/HPRC/NA20752/...                    4   \n",
       "\n",
       "   total_coverage                              coverage  \\\n",
       "0            58.8    [12.1, 11.7, 1.9, 9.4, 11.7, 12.0]   \n",
       "1            59.9  [6.4, 8.1, 6.4, 11.7, 8.8, 9.5, 9.0]   \n",
       "2            28.4                                [28.4]   \n",
       "3            63.8        [10.9, 11.6, 11.4, 11.0, 18.9]   \n",
       "4            65.5              [15.5, 15.1, 13.6, 21.3]   \n",
       "\n",
       "  number_of_cores_per_task                                   instrument_model  \\\n",
       "0                       10  ['Sequel II', 'Sequel II', 'Sequel II', 'Seque...   \n",
       "1                        9  ['Sequel II', 'Sequel II', 'Sequel II', 'Seque...   \n",
       "2                       64                                          ['Revio']   \n",
       "3                       12      ['Revio', 'Revio', 'Revio', 'Revio', 'Revio']   \n",
       "4                       16   ['Sequel II', 'Sequel II', 'Sequel II', 'Revio']   \n",
       "\n",
       "  hmm_flagger_window_size                              hmm_flagger_alpha_tsv  \n",
       "0                   16000  https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "1                   16000  https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "2                   16000  https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "3                   16000  https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "4                   16000  https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(f\"Number of rows in hifi full reads table (jan 12 2025 version) with censat batch2: {len(hifi_full_data_table_batch2)}\")\n",
    "hifi_full_data_table_batch2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of rows in hifi deepconsensus reads table with censat batch2: 3\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>asm_hap1</th>\n",
       "      <th>asm_hap2</th>\n",
       "      <th>cenSatAnnotations_hap1</th>\n",
       "      <th>cenSatAnnotations_hap2</th>\n",
       "      <th>censat_diploid_bed</th>\n",
       "      <th>additional_annotations_array</th>\n",
       "      <th>read_files</th>\n",
       "      <th>number_of_read_files</th>\n",
       "      <th>total_coverage</th>\n",
       "      <th>coverage</th>\n",
       "      <th>mapper_preset</th>\n",
       "      <th>kmer_size</th>\n",
       "      <th>number_of_cores_per_task</th>\n",
       "      <th>hmm_flagger_window_size</th>\n",
       "      <th>hmm_flagger_alpha_tsv</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG02145</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02145_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02145_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/submissions/3A25CF8A-...</td>\n",
       "      <td>4</td>\n",
       "      <td>43.28</td>\n",
       "      <td>[11.21, 8.8, 9.77, 13.5]</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>16</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HG01243</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG01243_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG01243_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/submissions/3A25CF8A-...</td>\n",
       "      <td>3</td>\n",
       "      <td>42.18</td>\n",
       "      <td>[15.29, 13.1, 13.79]</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>21</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NA19240</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA19240_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA19240_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/submissions/3A25CF8A-...</td>\n",
       "      <td>3</td>\n",
       "      <td>35.24</td>\n",
       "      <td>[12.44, 11.3, 11.5]</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>21</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   sample_id                                           asm_hap1  \\\n",
       "4    HG02145  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "12   HG01243  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "13   NA19240  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "\n",
       "                                             asm_hap2  \\\n",
       "4   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "12  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "13  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "\n",
       "                               cenSatAnnotations_hap1  \\\n",
       "4   /private/groups/hprc/qc/batch2/censat/HG02145_...   \n",
       "12  /private/groups/hprc/qc/batch2/censat/HG01243_...   \n",
       "13  /private/groups/hprc/qc/batch2/censat/NA19240_...   \n",
       "\n",
       "                               cenSatAnnotations_hap2  \\\n",
       "4   /private/groups/hprc/qc/batch2/censat/HG02145_...   \n",
       "12  /private/groups/hprc/qc/batch2/censat/HG01243_...   \n",
       "13  /private/groups/hprc/qc/batch2/censat/NA19240_...   \n",
       "\n",
       "                                   censat_diploid_bed  \\\n",
       "4   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "12  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "13  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "\n",
       "                         additional_annotations_array  \\\n",
       "4   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "12  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "13  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "\n",
       "                                           read_files  number_of_read_files  \\\n",
       "4   ['s3://human-pangenomics/submissions/3A25CF8A-...                     4   \n",
       "12  ['s3://human-pangenomics/submissions/3A25CF8A-...                     3   \n",
       "13  ['s3://human-pangenomics/submissions/3A25CF8A-...                     3   \n",
       "\n",
       "    total_coverage                  coverage mapper_preset  kmer_size  \\\n",
       "4            43.28  [11.21, 8.8, 9.77, 13.5]       lr:hqae         25   \n",
       "12           42.18      [15.29, 13.1, 13.79]       lr:hqae         25   \n",
       "13           35.24       [12.44, 11.3, 11.5]       lr:hqae         25   \n",
       "\n",
       "    number_of_cores_per_task  hmm_flagger_window_size  \\\n",
       "4                         16                    16000   \n",
       "12                        21                    16000   \n",
       "13                        21                    16000   \n",
       "\n",
       "                                hmm_flagger_alpha_tsv  \n",
       "4   https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "12  https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "13  https://raw.githubusercontent.com/mobinasri/fl...  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(f\"Number of rows in hifi deepconsensus reads table with censat batch2: {len(hifi_dc_data_table_batch2)}\")\n",
    "hifi_dc_data_table_batch2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# since we are going to use the columns that end with \"_downsampled\" for creating WDL json files\n",
    "# and hifi DeepConsensus tables didnot have those I will make them here\n",
    "hifi_dc_data_table_batch2[\"read_files_downsampled\"] = hifi_dc_data_table_batch2[\"read_files\"]\n",
    "hifi_dc_data_table_batch2[\"total_coverage_downsampled\"] = hifi_dc_data_table_batch2[\"total_coverage\"]\n",
    "hifi_dc_data_table_batch2[\"number_of_read_files_downsampled\"] = hifi_dc_data_table_batch2[\"number_of_read_files\"]\n",
    "hifi_dc_data_table_batch2[\"number_of_cores_per_task_downsampled\"] = hifi_dc_data_table_batch2[\"number_of_cores_per_task\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>asm_hap1</th>\n",
       "      <th>asm_hap2</th>\n",
       "      <th>cenSatAnnotations_hap1</th>\n",
       "      <th>cenSatAnnotations_hap2</th>\n",
       "      <th>censat_diploid_bed</th>\n",
       "      <th>additional_annotations_array</th>\n",
       "      <th>read_files_downsampled</th>\n",
       "      <th>number_of_read_files_downsampled</th>\n",
       "      <th>total_coverage_downsampled</th>\n",
       "      <th>...</th>\n",
       "      <th>mapper_preset</th>\n",
       "      <th>kmer_size</th>\n",
       "      <th>read_files</th>\n",
       "      <th>number_of_read_files</th>\n",
       "      <th>total_coverage</th>\n",
       "      <th>coverage</th>\n",
       "      <th>number_of_cores_per_task</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>hmm_flagger_window_size</th>\n",
       "      <th>hmm_flagger_alpha_tsv</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG01975</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG01975_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG01975_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG01975/...</td>\n",
       "      <td>6</td>\n",
       "      <td>58.80</td>\n",
       "      <td>...</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG01975/...</td>\n",
       "      <td>6</td>\n",
       "      <td>58.80</td>\n",
       "      <td>[12.1, 11.7, 1.9, 9.4, 11.7, 12.0]</td>\n",
       "      <td>10</td>\n",
       "      <td>['Sequel II', 'Sequel II', 'Sequel II', 'Seque...</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG02602</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02602_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02602_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02602/...</td>\n",
       "      <td>7</td>\n",
       "      <td>59.90</td>\n",
       "      <td>...</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02602/...</td>\n",
       "      <td>7</td>\n",
       "      <td>59.90</td>\n",
       "      <td>[6.4, 8.1, 6.4, 11.7, 8.8, 9.5, 9.0]</td>\n",
       "      <td>9</td>\n",
       "      <td>['Sequel II', 'Sequel II', 'Sequel II', 'Seque...</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG04187</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG04187_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG04187_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG04187/...</td>\n",
       "      <td>1</td>\n",
       "      <td>28.40</td>\n",
       "      <td>...</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG04187/...</td>\n",
       "      <td>1</td>\n",
       "      <td>28.40</td>\n",
       "      <td>[28.4]</td>\n",
       "      <td>64</td>\n",
       "      <td>['Revio']</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NA18879</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA18879_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA18879_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/NA18879/...</td>\n",
       "      <td>5</td>\n",
       "      <td>63.80</td>\n",
       "      <td>...</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/NA18879/...</td>\n",
       "      <td>5</td>\n",
       "      <td>63.80</td>\n",
       "      <td>[10.9, 11.6, 11.4, 11.0, 18.9]</td>\n",
       "      <td>12</td>\n",
       "      <td>['Revio', 'Revio', 'Revio', 'Revio', 'Revio']</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NA20752</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA20752_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA20752_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/NA20752/...</td>\n",
       "      <td>4</td>\n",
       "      <td>65.50</td>\n",
       "      <td>...</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/NA20752/...</td>\n",
       "      <td>4</td>\n",
       "      <td>65.50</td>\n",
       "      <td>[15.5, 15.1, 13.6, 21.3]</td>\n",
       "      <td>16</td>\n",
       "      <td>['Sequel II', 'Sequel II', 'Sequel II', 'Revio']</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HG02738</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02738_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02738_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02738/...</td>\n",
       "      <td>6</td>\n",
       "      <td>58.20</td>\n",
       "      <td>...</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02738/...</td>\n",
       "      <td>6</td>\n",
       "      <td>58.20</td>\n",
       "      <td>[7.2, 7.8, 7.4, 8.1, 15.8, 11.9]</td>\n",
       "      <td>10</td>\n",
       "      <td>['Sequel II', 'Sequel II', 'Sequel II', 'Seque...</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HG00706</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG00706_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG00706_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG00706/...</td>\n",
       "      <td>5</td>\n",
       "      <td>61.10</td>\n",
       "      <td>...</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG00706/...</td>\n",
       "      <td>5</td>\n",
       "      <td>61.10</td>\n",
       "      <td>[8.9, 7.8, 9.5, 9.5, 25.4]</td>\n",
       "      <td>12</td>\n",
       "      <td>['Sequel II', 'Sequel II', 'Sequel II', 'Seque...</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>HG03816</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG03816_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG03816_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG03816/...</td>\n",
       "      <td>5</td>\n",
       "      <td>59.90</td>\n",
       "      <td>...</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG03816/...</td>\n",
       "      <td>5</td>\n",
       "      <td>59.90</td>\n",
       "      <td>[7.9, 8.5, 9.0, 8.6, 25.9]</td>\n",
       "      <td>12</td>\n",
       "      <td>['Sequel II', 'Sequel II', 'Sequel II', 'Seque...</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NA19159</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA19159_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA19159_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/NA19159/...</td>\n",
       "      <td>4</td>\n",
       "      <td>52.54</td>\n",
       "      <td>...</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/NA19159/...</td>\n",
       "      <td>4</td>\n",
       "      <td>52.54</td>\n",
       "      <td>[14.9, 14.9, 15.1, 7.64]</td>\n",
       "      <td>16</td>\n",
       "      <td>['Sequel II', 'Sequel II', 'Sequel II', 'Revio']</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HG02392</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02392_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02392_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02392/...</td>\n",
       "      <td>4</td>\n",
       "      <td>60.60</td>\n",
       "      <td>...</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02392/...</td>\n",
       "      <td>5</td>\n",
       "      <td>62.20</td>\n",
       "      <td>[21.4, 22.7, 12.2, 1.6, 4.3]</td>\n",
       "      <td>12</td>\n",
       "      <td>['Revio', 'Revio', 'Revio', 'Revio', 'Revio']</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>HG02514</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02514_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02514_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02514/...</td>\n",
       "      <td>6</td>\n",
       "      <td>54.64</td>\n",
       "      <td>...</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02514/...</td>\n",
       "      <td>6</td>\n",
       "      <td>54.64</td>\n",
       "      <td>[5.1, 10.2, 4.5, 10.0, 5.1, 19.74]</td>\n",
       "      <td>10</td>\n",
       "      <td>['Sequel II', 'Sequel II', 'Sequel II', 'Seque...</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>HG02841</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02841_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02841_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02841/...</td>\n",
       "      <td>8</td>\n",
       "      <td>62.36</td>\n",
       "      <td>...</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02841/...</td>\n",
       "      <td>8</td>\n",
       "      <td>62.36</td>\n",
       "      <td>[10.1, 10.3, 10.2, 2.9, 3.6, 8.52, 8.38, 8.36]</td>\n",
       "      <td>8</td>\n",
       "      <td>['Sequel II', 'Sequel II', 'Sequel II', 'Seque...</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HG02984</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02984_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02984_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02984/...</td>\n",
       "      <td>7</td>\n",
       "      <td>58.74</td>\n",
       "      <td>...</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02984/...</td>\n",
       "      <td>7</td>\n",
       "      <td>58.74</td>\n",
       "      <td>[7.0, 8.4, 7.3, 6.4, 0.6, 5.6, 23.44]</td>\n",
       "      <td>9</td>\n",
       "      <td>['Sequel II', 'Sequel II', 'Sequel II', 'Seque...</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>HG03050</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG03050_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG03050_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG03050/...</td>\n",
       "      <td>6</td>\n",
       "      <td>47.17</td>\n",
       "      <td>...</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG03050/...</td>\n",
       "      <td>6</td>\n",
       "      <td>47.17</td>\n",
       "      <td>[11.5, 10.1, 3.0, 8.6, 3.5, 10.47]</td>\n",
       "      <td>10</td>\n",
       "      <td>['Sequel II', 'Sequel II', 'Sequel II', 'Seque...</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>NA20799</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA20799_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA20799_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/NA20799/...</td>\n",
       "      <td>6</td>\n",
       "      <td>60.93</td>\n",
       "      <td>...</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/NA20799/...</td>\n",
       "      <td>6</td>\n",
       "      <td>60.93</td>\n",
       "      <td>[11.6, 11.5, 10.8, 5.8, 12.09, 9.14]</td>\n",
       "      <td>10</td>\n",
       "      <td>['Sequel II', 'Sequel II', 'Sequel II', 'Seque...</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>HG02145</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02145_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG02145_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/submissions/3A25CF8A-...</td>\n",
       "      <td>4</td>\n",
       "      <td>43.28</td>\n",
       "      <td>...</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/submissions/3A25CF8A-...</td>\n",
       "      <td>4</td>\n",
       "      <td>43.28</td>\n",
       "      <td>[11.21, 8.8, 9.77, 13.5]</td>\n",
       "      <td>16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>HG01243</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG01243_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/HG01243_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/submissions/3A25CF8A-...</td>\n",
       "      <td>3</td>\n",
       "      <td>42.18</td>\n",
       "      <td>...</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/submissions/3A25CF8A-...</td>\n",
       "      <td>3</td>\n",
       "      <td>42.18</td>\n",
       "      <td>[15.29, 13.1, 13.79]</td>\n",
       "      <td>21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>NA19240</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA19240_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch2/censat/NA19240_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/submissions/3A25CF8A-...</td>\n",
       "      <td>3</td>\n",
       "      <td>35.24</td>\n",
       "      <td>...</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>['s3://human-pangenomics/submissions/3A25CF8A-...</td>\n",
       "      <td>3</td>\n",
       "      <td>35.24</td>\n",
       "      <td>[12.44, 11.3, 11.5]</td>\n",
       "      <td>21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   sample_id                                           asm_hap1  \\\n",
       "0    HG01975  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "1    HG02602  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "2    HG04187  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "3    NA18879  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "4    NA20752  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "5    HG02738  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "6    HG00706  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "7    HG03816  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "8    NA19159  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "9    HG02392  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "10   HG02514  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "11   HG02841  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "12   HG02984  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "13   HG03050  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "14   NA20799  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "15   HG02145  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "16   HG01243  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "17   NA19240  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "\n",
       "                                             asm_hap2  \\\n",
       "0   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "1   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "2   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "3   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "4   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "5   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "6   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "7   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "8   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "9   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "10  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "11  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "12  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "13  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "14  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "15  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "16  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "17  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "\n",
       "                               cenSatAnnotations_hap1  \\\n",
       "0   /private/groups/hprc/qc/batch2/censat/HG01975_...   \n",
       "1   /private/groups/hprc/qc/batch2/censat/HG02602_...   \n",
       "2   /private/groups/hprc/qc/batch2/censat/HG04187_...   \n",
       "3   /private/groups/hprc/qc/batch2/censat/NA18879_...   \n",
       "4   /private/groups/hprc/qc/batch2/censat/NA20752_...   \n",
       "5   /private/groups/hprc/qc/batch2/censat/HG02738_...   \n",
       "6   /private/groups/hprc/qc/batch2/censat/HG00706_...   \n",
       "7   /private/groups/hprc/qc/batch2/censat/HG03816_...   \n",
       "8   /private/groups/hprc/qc/batch2/censat/NA19159_...   \n",
       "9   /private/groups/hprc/qc/batch2/censat/HG02392_...   \n",
       "10  /private/groups/hprc/qc/batch2/censat/HG02514_...   \n",
       "11  /private/groups/hprc/qc/batch2/censat/HG02841_...   \n",
       "12  /private/groups/hprc/qc/batch2/censat/HG02984_...   \n",
       "13  /private/groups/hprc/qc/batch2/censat/HG03050_...   \n",
       "14  /private/groups/hprc/qc/batch2/censat/NA20799_...   \n",
       "15  /private/groups/hprc/qc/batch2/censat/HG02145_...   \n",
       "16  /private/groups/hprc/qc/batch2/censat/HG01243_...   \n",
       "17  /private/groups/hprc/qc/batch2/censat/NA19240_...   \n",
       "\n",
       "                               cenSatAnnotations_hap2  \\\n",
       "0   /private/groups/hprc/qc/batch2/censat/HG01975_...   \n",
       "1   /private/groups/hprc/qc/batch2/censat/HG02602_...   \n",
       "2   /private/groups/hprc/qc/batch2/censat/HG04187_...   \n",
       "3   /private/groups/hprc/qc/batch2/censat/NA18879_...   \n",
       "4   /private/groups/hprc/qc/batch2/censat/NA20752_...   \n",
       "5   /private/groups/hprc/qc/batch2/censat/HG02738_...   \n",
       "6   /private/groups/hprc/qc/batch2/censat/HG00706_...   \n",
       "7   /private/groups/hprc/qc/batch2/censat/HG03816_...   \n",
       "8   /private/groups/hprc/qc/batch2/censat/NA19159_...   \n",
       "9   /private/groups/hprc/qc/batch2/censat/HG02392_...   \n",
       "10  /private/groups/hprc/qc/batch2/censat/HG02514_...   \n",
       "11  /private/groups/hprc/qc/batch2/censat/HG02841_...   \n",
       "12  /private/groups/hprc/qc/batch2/censat/HG02984_...   \n",
       "13  /private/groups/hprc/qc/batch2/censat/HG03050_...   \n",
       "14  /private/groups/hprc/qc/batch2/censat/NA20799_...   \n",
       "15  /private/groups/hprc/qc/batch2/censat/HG02145_...   \n",
       "16  /private/groups/hprc/qc/batch2/censat/HG01243_...   \n",
       "17  /private/groups/hprc/qc/batch2/censat/NA19240_...   \n",
       "\n",
       "                                   censat_diploid_bed  \\\n",
       "0   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "1   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "2   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "3   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "4   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "5   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "6   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "7   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "8   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "9   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "10  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "11  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "12  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "13  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "14  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "15  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "16  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "17  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "\n",
       "                         additional_annotations_array  \\\n",
       "0   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "1   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "2   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "3   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "4   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "5   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "6   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "7   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "8   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "9   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "10  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "11  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "12  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "13  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "14  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "15  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "16  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "17  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "\n",
       "                               read_files_downsampled  \\\n",
       "0   ['s3://human-pangenomics/working/HPRC/HG01975/...   \n",
       "1   ['s3://human-pangenomics/working/HPRC/HG02602/...   \n",
       "2   ['s3://human-pangenomics/working/HPRC/HG04187/...   \n",
       "3   ['s3://human-pangenomics/working/HPRC/NA18879/...   \n",
       "4   ['s3://human-pangenomics/working/HPRC/NA20752/...   \n",
       "5   ['s3://human-pangenomics/working/HPRC/HG02738/...   \n",
       "6   ['s3://human-pangenomics/working/HPRC/HG00706/...   \n",
       "7   ['s3://human-pangenomics/working/HPRC/HG03816/...   \n",
       "8   ['s3://human-pangenomics/working/HPRC/NA19159/...   \n",
       "9   ['s3://human-pangenomics/working/HPRC/HG02392/...   \n",
       "10  ['s3://human-pangenomics/working/HPRC/HG02514/...   \n",
       "11  ['s3://human-pangenomics/working/HPRC/HG02841/...   \n",
       "12  ['s3://human-pangenomics/working/HPRC/HG02984/...   \n",
       "13  ['s3://human-pangenomics/working/HPRC/HG03050/...   \n",
       "14  ['s3://human-pangenomics/working/HPRC/NA20799/...   \n",
       "15  ['s3://human-pangenomics/submissions/3A25CF8A-...   \n",
       "16  ['s3://human-pangenomics/submissions/3A25CF8A-...   \n",
       "17  ['s3://human-pangenomics/submissions/3A25CF8A-...   \n",
       "\n",
       "    number_of_read_files_downsampled  total_coverage_downsampled  \\\n",
       "0                                  6                       58.80   \n",
       "1                                  7                       59.90   \n",
       "2                                  1                       28.40   \n",
       "3                                  5                       63.80   \n",
       "4                                  4                       65.50   \n",
       "5                                  6                       58.20   \n",
       "6                                  5                       61.10   \n",
       "7                                  5                       59.90   \n",
       "8                                  4                       52.54   \n",
       "9                                  4                       60.60   \n",
       "10                                 6                       54.64   \n",
       "11                                 8                       62.36   \n",
       "12                                 7                       58.74   \n",
       "13                                 6                       47.17   \n",
       "14                                 6                       60.93   \n",
       "15                                 4                       43.28   \n",
       "16                                 3                       42.18   \n",
       "17                                 3                       35.24   \n",
       "\n",
       "                          ...                          mapper_preset  \\\n",
       "0                         ...                                lr:hqae   \n",
       "1                         ...                                lr:hqae   \n",
       "2                         ...                                lr:hqae   \n",
       "3                         ...                                lr:hqae   \n",
       "4                         ...                                lr:hqae   \n",
       "5                         ...                                lr:hqae   \n",
       "6                         ...                                lr:hqae   \n",
       "7                         ...                                lr:hqae   \n",
       "8                         ...                                lr:hqae   \n",
       "9                         ...                                lr:hqae   \n",
       "10                        ...                                lr:hqae   \n",
       "11                        ...                                lr:hqae   \n",
       "12                        ...                                lr:hqae   \n",
       "13                        ...                                lr:hqae   \n",
       "14                        ...                                lr:hqae   \n",
       "15                        ...                                lr:hqae   \n",
       "16                        ...                                lr:hqae   \n",
       "17                        ...                                lr:hqae   \n",
       "\n",
       "   kmer_size                                         read_files  \\\n",
       "0         25  ['s3://human-pangenomics/working/HPRC/HG01975/...   \n",
       "1         25  ['s3://human-pangenomics/working/HPRC/HG02602/...   \n",
       "2         25  ['s3://human-pangenomics/working/HPRC/HG04187/...   \n",
       "3         25  ['s3://human-pangenomics/working/HPRC/NA18879/...   \n",
       "4         25  ['s3://human-pangenomics/working/HPRC/NA20752/...   \n",
       "5         25  ['s3://human-pangenomics/working/HPRC/HG02738/...   \n",
       "6         25  ['s3://human-pangenomics/working/HPRC/HG00706/...   \n",
       "7         25  ['s3://human-pangenomics/working/HPRC/HG03816/...   \n",
       "8         25  ['s3://human-pangenomics/working/HPRC/NA19159/...   \n",
       "9         25  ['s3://human-pangenomics/working/HPRC/HG02392/...   \n",
       "10        25  ['s3://human-pangenomics/working/HPRC/HG02514/...   \n",
       "11        25  ['s3://human-pangenomics/working/HPRC/HG02841/...   \n",
       "12        25  ['s3://human-pangenomics/working/HPRC/HG02984/...   \n",
       "13        25  ['s3://human-pangenomics/working/HPRC/HG03050/...   \n",
       "14        25  ['s3://human-pangenomics/working/HPRC/NA20799/...   \n",
       "15        25  ['s3://human-pangenomics/submissions/3A25CF8A-...   \n",
       "16        25  ['s3://human-pangenomics/submissions/3A25CF8A-...   \n",
       "17        25  ['s3://human-pangenomics/submissions/3A25CF8A-...   \n",
       "\n",
       "   number_of_read_files  total_coverage  \\\n",
       "0                     6           58.80   \n",
       "1                     7           59.90   \n",
       "2                     1           28.40   \n",
       "3                     5           63.80   \n",
       "4                     4           65.50   \n",
       "5                     6           58.20   \n",
       "6                     5           61.10   \n",
       "7                     5           59.90   \n",
       "8                     4           52.54   \n",
       "9                     5           62.20   \n",
       "10                    6           54.64   \n",
       "11                    8           62.36   \n",
       "12                    7           58.74   \n",
       "13                    6           47.17   \n",
       "14                    6           60.93   \n",
       "15                    4           43.28   \n",
       "16                    3           42.18   \n",
       "17                    3           35.24   \n",
       "\n",
       "                                          coverage number_of_cores_per_task  \\\n",
       "0               [12.1, 11.7, 1.9, 9.4, 11.7, 12.0]                       10   \n",
       "1             [6.4, 8.1, 6.4, 11.7, 8.8, 9.5, 9.0]                        9   \n",
       "2                                           [28.4]                       64   \n",
       "3                   [10.9, 11.6, 11.4, 11.0, 18.9]                       12   \n",
       "4                         [15.5, 15.1, 13.6, 21.3]                       16   \n",
       "5                 [7.2, 7.8, 7.4, 8.1, 15.8, 11.9]                       10   \n",
       "6                       [8.9, 7.8, 9.5, 9.5, 25.4]                       12   \n",
       "7                       [7.9, 8.5, 9.0, 8.6, 25.9]                       12   \n",
       "8                         [14.9, 14.9, 15.1, 7.64]                       16   \n",
       "9                     [21.4, 22.7, 12.2, 1.6, 4.3]                       12   \n",
       "10              [5.1, 10.2, 4.5, 10.0, 5.1, 19.74]                       10   \n",
       "11  [10.1, 10.3, 10.2, 2.9, 3.6, 8.52, 8.38, 8.36]                        8   \n",
       "12           [7.0, 8.4, 7.3, 6.4, 0.6, 5.6, 23.44]                        9   \n",
       "13              [11.5, 10.1, 3.0, 8.6, 3.5, 10.47]                       10   \n",
       "14            [11.6, 11.5, 10.8, 5.8, 12.09, 9.14]                       10   \n",
       "15                        [11.21, 8.8, 9.77, 13.5]                       16   \n",
       "16                            [15.29, 13.1, 13.79]                       21   \n",
       "17                             [12.44, 11.3, 11.5]                       21   \n",
       "\n",
       "                                     instrument_model hmm_flagger_window_size  \\\n",
       "0   ['Sequel II', 'Sequel II', 'Sequel II', 'Seque...                   16000   \n",
       "1   ['Sequel II', 'Sequel II', 'Sequel II', 'Seque...                   16000   \n",
       "2                                           ['Revio']                   16000   \n",
       "3       ['Revio', 'Revio', 'Revio', 'Revio', 'Revio']                   16000   \n",
       "4    ['Sequel II', 'Sequel II', 'Sequel II', 'Revio']                   16000   \n",
       "5   ['Sequel II', 'Sequel II', 'Sequel II', 'Seque...                   16000   \n",
       "6   ['Sequel II', 'Sequel II', 'Sequel II', 'Seque...                   16000   \n",
       "7   ['Sequel II', 'Sequel II', 'Sequel II', 'Seque...                   16000   \n",
       "8    ['Sequel II', 'Sequel II', 'Sequel II', 'Revio']                   16000   \n",
       "9       ['Revio', 'Revio', 'Revio', 'Revio', 'Revio']                   16000   \n",
       "10  ['Sequel II', 'Sequel II', 'Sequel II', 'Seque...                   16000   \n",
       "11  ['Sequel II', 'Sequel II', 'Sequel II', 'Seque...                   16000   \n",
       "12  ['Sequel II', 'Sequel II', 'Sequel II', 'Seque...                   16000   \n",
       "13  ['Sequel II', 'Sequel II', 'Sequel II', 'Seque...                   16000   \n",
       "14  ['Sequel II', 'Sequel II', 'Sequel II', 'Seque...                   16000   \n",
       "15                                                NaN                   16000   \n",
       "16                                                NaN                   16000   \n",
       "17                                                NaN                   16000   \n",
       "\n",
       "                                hmm_flagger_alpha_tsv  \n",
       "0   https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "1   https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "2   https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "3   https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "4   https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "5   https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "6   https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "7   https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "8   https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "9   https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "10  https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "11  https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "12  https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "13  https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "14  https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "15  https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "16  https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "17  https://raw.githubusercontent.com/mobinasri/fl...  \n",
       "\n",
       "[18 rows x 21 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Concatenate vertically\n",
    "merged_hifi_data_table_batch2 = pd.concat([hifi_full_data_table_batch2, hifi_dc_data_table_batch2], sort=False)\n",
    "merged_hifi_data_table_batch2.index = np.arange(len(merged_hifi_data_table_batch2))\n",
    "merged_hifi_data_table_batch2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "These 2 samples don't have HiFi reads for censat batch 2: \n",
      "\n",
      "HG00733\n",
      "HG005\n"
     ]
    }
   ],
   "source": [
    "missing_samples = set(censat_table_diploid_batch2['sample_id']).difference(merged_hifi_data_table_batch2['sample_id'])\n",
    "print(f\"These {len(missing_samples)} samples don't have HiFi reads for censat batch 2: \\n\")\n",
    "print(\"\\n\".join(missing_samples))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "os.makedirs(\"ont\", exist_ok=True)\n",
    "merged_ont_data_table_batch2.to_csv('ont/hmm_flagger_ont_data_table.csv', index=False)\n",
    "\n",
    "os.makedirs(\"hifi\", exist_ok=True)\n",
    "merged_hifi_data_table_batch2.to_csv('hifi/hmm_flagger_hifi_data_table.csv', index=False)"
   ]
  }
 ],
 "metadata": {
  "colab": {
   "provenance": []
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
