{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "id": "X_i8u8SG-3yd"
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import sys\n",
    "import os\n",
    "import re"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### This notebook:\n",
    "- Opens `batch1_censat_outputs_done.csv` and `hprc_year1_sample_metadata.txt`\n",
    "- Makes diploid censat bed files and add links to the final tables\n",
    "- Adds a column to the final tables which keeps if the sample is Year1 or not\n",
    "- Puts Year1 samples on top of the table so that they will be run first\n",
    "- Takes read paths from tables `/private/groups/hprc/qc_hmm_flagger/hprc_intermediate_assembly/assembly_qc/batch1/hmm_flagger/read_tables/ont_reads_table.csv` and `/private/groups/hprc/qc_hmm_flagger/hprc_intermediate_assembly/assembly_qc/batch1/hmm_flagger/read_tables/hifi_reads_table.csv`\n",
    "- Makes separate data tables for HiFi and ONT runs (both will contain diploid censat bed files)\n",
    "- Saves the final data tables in `hifi/` and `ont/` subdirectories and they will be used for creating input json files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "batch1_censat_outputs_done.csv\thprc_year1_sample_metadata.txt\t    read_tables\r\n",
      "diploid_censat_beds\t\tmake_hmm_flagger_data_tables.ipynb\r\n",
      "hifi\t\t\t\tont\r\n"
     ]
    }
   ],
   "source": [
    "!ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of rows: 388\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>asm</th>\n",
       "      <th>sequence_id_key</th>\n",
       "      <th>as_sf_bed</th>\n",
       "      <th>as_hor_bed</th>\n",
       "      <th>as_strand_bed</th>\n",
       "      <th>as_hor_sf_bed</th>\n",
       "      <th>rmRmskAlignBigBed</th>\n",
       "      <th>rmRmskBigBed</th>\n",
       "      <th>repeatMaskerTarGZ</th>\n",
       "      <th>rmOutFile</th>\n",
       "      <th>rmFinalMaskedFasta</th>\n",
       "      <th>rmRmskAlignBed</th>\n",
       "      <th>rmRmskBed</th>\n",
       "      <th>rmBed</th>\n",
       "      <th>centromeres</th>\n",
       "      <th>cenSatStrand</th>\n",
       "      <th>cenSatAnnotations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG00408_hap1</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG00408_hap2</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG00597_hap1</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG00597_hap2</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG01192_hap1</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG01192_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG01192_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG01192_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG01192_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG01192_...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG01192_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG01192_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG01192_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG01192_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG01192_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG01192_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG01192_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG01192_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG01192_...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      sample_id                                                asm  \\\n",
       "0  HG00408_hap1  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "1  HG00408_hap2  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "2  HG00597_hap1  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "3  HG00597_hap2  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "4  HG01192_hap1  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "\n",
       "                                     sequence_id_key  \\\n",
       "0  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "1  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "2  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "3  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "4  /private/groups/hprc/qc/batch1/censat/HG01192_...   \n",
       "\n",
       "                                           as_sf_bed  \\\n",
       "0  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "1  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "2  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "3  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "4  /private/groups/hprc/qc/batch1/censat/HG01192_...   \n",
       "\n",
       "                                          as_hor_bed  \\\n",
       "0  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "1  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "2  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "3  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "4  /private/groups/hprc/qc/batch1/censat/HG01192_...   \n",
       "\n",
       "                                       as_strand_bed  \\\n",
       "0  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "1  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "2  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "3  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "4  /private/groups/hprc/qc/batch1/censat/HG01192_...   \n",
       "\n",
       "                                       as_hor_sf_bed  rmRmskAlignBigBed  \\\n",
       "0  /private/groups/hprc/qc/batch1/censat/HG00408_...                NaN   \n",
       "1  /private/groups/hprc/qc/batch1/censat/HG00408_...                NaN   \n",
       "2  /private/groups/hprc/qc/batch1/censat/HG00597_...                NaN   \n",
       "3  /private/groups/hprc/qc/batch1/censat/HG00597_...                NaN   \n",
       "4  /private/groups/hprc/qc/batch1/censat/HG01192_...                NaN   \n",
       "\n",
       "   rmRmskBigBed                                  repeatMaskerTarGZ  \\\n",
       "0           NaN  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "1           NaN  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "2           NaN  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "3           NaN  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "4           NaN  /private/groups/hprc/qc/batch1/censat/HG01192_...   \n",
       "\n",
       "                                           rmOutFile  \\\n",
       "0  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "1  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "2  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "3  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "4  /private/groups/hprc/qc/batch1/censat/HG01192_...   \n",
       "\n",
       "                                  rmFinalMaskedFasta  \\\n",
       "0  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "1  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "2  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "3  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "4  /private/groups/hprc/qc/batch1/censat/HG01192_...   \n",
       "\n",
       "                                      rmRmskAlignBed  \\\n",
       "0  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "1  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "2  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "3  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "4  /private/groups/hprc/qc/batch1/censat/HG01192_...   \n",
       "\n",
       "                                           rmRmskBed  \\\n",
       "0  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "1  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "2  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "3  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "4  /private/groups/hprc/qc/batch1/censat/HG01192_...   \n",
       "\n",
       "                                               rmBed  \\\n",
       "0  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "1  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "2  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "3  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "4  /private/groups/hprc/qc/batch1/censat/HG01192_...   \n",
       "\n",
       "                                         centromeres  \\\n",
       "0  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "1  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "2  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "3  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "4  /private/groups/hprc/qc/batch1/censat/HG01192_...   \n",
       "\n",
       "                                        cenSatStrand  \\\n",
       "0  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "1  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "2  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "3  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "4  /private/groups/hprc/qc/batch1/censat/HG01192_...   \n",
       "\n",
       "                                   cenSatAnnotations  \n",
       "0  /private/groups/hprc/qc/batch1/censat/HG00408_...  \n",
       "1  /private/groups/hprc/qc/batch1/censat/HG00408_...  \n",
       "2  /private/groups/hprc/qc/batch1/censat/HG00597_...  \n",
       "3  /private/groups/hprc/qc/batch1/censat/HG00597_...  \n",
       "4  /private/groups/hprc/qc/batch1/censat/HG01192_...  "
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "censat_table = pd.read_csv('batch1_censat_outputs_done.csv')\n",
    "print(\"Number of rows:\", len(censat_table))\n",
    "censat_table.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>asm_hap1</th>\n",
       "      <th>asm_hap2</th>\n",
       "      <th>cenSatAnnotations_hap1</th>\n",
       "      <th>cenSatAnnotations_hap2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG00408</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG00597</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG01192</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG01192_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG01192_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG01261</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG01261_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG01261_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG02015</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG02015_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG02015_...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  sample_id                                           asm_hap1  \\\n",
       "0   HG00408  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "1   HG00597  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "2   HG01192  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "3   HG01261  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "4   HG02015  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "\n",
       "                                            asm_hap2  \\\n",
       "0  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "1  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "2  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "3  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "4  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "\n",
       "                              cenSatAnnotations_hap1  \\\n",
       "0  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "1  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "2  /private/groups/hprc/qc/batch1/censat/HG01192_...   \n",
       "3  /private/groups/hprc/qc/batch1/censat/HG01261_...   \n",
       "4  /private/groups/hprc/qc/batch1/censat/HG02015_...   \n",
       "\n",
       "                              cenSatAnnotations_hap2  \n",
       "0  /private/groups/hprc/qc/batch1/censat/HG00408_...  \n",
       "1  /private/groups/hprc/qc/batch1/censat/HG00597_...  \n",
       "2  /private/groups/hprc/qc/batch1/censat/HG01192_...  \n",
       "3  /private/groups/hprc/qc/batch1/censat/HG01261_...  \n",
       "4  /private/groups/hprc/qc/batch1/censat/HG02015_...  "
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Remove \"_hap1\" or \"_hap2\" from 'sample_id' and create a new column 'sample_name'\n",
    "censat_table[\"sample_name\"] = censat_table[\"sample_id\"].str.replace(r\"_hap[12]$\", \"\", regex=True)\n",
    "\n",
    "# make two tables one for hap1 and one for hap2\n",
    "censat_table_hap1 = censat_table[censat_table['sample_id'].str.endswith('_hap1', na=False)]\n",
    "censat_table_hap2 = censat_table[censat_table['sample_id'].str.endswith('_hap2', na=False)]\n",
    "\n",
    "# Merging the DataFrames on 'sample_name'\n",
    "censat_table_diploid = pd.merge(censat_table_hap1,\n",
    "                                censat_table_hap2,\n",
    "                                on='sample_name',\n",
    "                                suffixes=('_hap1', '_hap2'))\n",
    "censat_table_diploid = censat_table_diploid[[\"sample_name\", \n",
    "                                             \"asm_hap1\", \n",
    "                                             \"asm_hap2\",\n",
    "                                             \"cenSatAnnotations_hap1\",\n",
    "                                             \"cenSatAnnotations_hap2\"]]\n",
    "\n",
    "censat_table_diploid = censat_table_diploid.rename(columns={\"sample_name\":\"sample_id\"})\n",
    "censat_table_diploid.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Merged HG00408\n",
      "Merged HG00597\n",
      "Merged HG01192\n",
      "Merged HG01261\n",
      "Merged HG02015\n",
      "Merged HG02056\n",
      "Merged HG02129\n",
      "Merged HG02258\n",
      "Merged HG03834\n",
      "Merged HG00609\n",
      "Merged HG00642\n",
      "Merged HG00738\n",
      "Merged HG01099\n",
      "Merged HG01255\n",
      "Merged HG01346\n",
      "Merged HG01433\n",
      "Merged HG01496\n",
      "Merged HG01884\n",
      "Merged HG01981\n",
      "Merged HG01993\n",
      "Merged HG02004\n",
      "Merged HG02083\n",
      "Merged HG02132\n",
      "Merged HG02280\n",
      "Merged HG02293\n",
      "Merged HG02300\n",
      "Merged HG02451\n",
      "Merged HG02523\n",
      "Merged HG02615\n",
      "Merged HG02647\n",
      "Merged HG02698\n",
      "Merged HG03654\n",
      "Merged HG03669\n",
      "Merged HG03688\n",
      "Merged HG03704\n",
      "Merged HG03710\n",
      "Merged HG03831\n",
      "Merged HG03927\n",
      "Merged HG04115\n",
      "Merged HG04184\n",
      "Merged HG04199\n",
      "Merged HG00621\n",
      "Merged HG00741\n",
      "Merged HG01106\n",
      "Merged HG01175\n",
      "Merged HG01258\n",
      "Merged HG01891\n",
      "Merged HG01952\n",
      "Merged HG02148\n",
      "Merged HG02486\n",
      "Merged HG02559\n",
      "Merged HG02572\n",
      "Merged HG02622\n",
      "Merged HG02886\n",
      "Merged HG03453\n",
      "Merged HG03540\n",
      "Merged HG00099\n",
      "Merged HG00140\n",
      "Merged HG00280\n",
      "Merged HG00323\n",
      "Merged HG02040\n",
      "Merged HG02165\n",
      "Merged HG02273\n",
      "Merged HG02922\n",
      "Merged HG02965\n",
      "Merged HG02976\n",
      "Merged HG03130\n",
      "Merged HG03195\n",
      "Merged HG03209\n",
      "Merged NA18522\n",
      "Merged NA18747\n",
      "Merged NA18971\n",
      "Merged NA18983\n",
      "Merged NA19043\n",
      "Merged NA20905\n",
      "Merged HG01530\n",
      "Merged HG01784\n",
      "Merged HG02391\n",
      "Merged HG03471\n",
      "Merged HG03742\n",
      "Merged NA18570\n",
      "Merged NA19087\n",
      "Merged NA19185\n",
      "Merged NA19338\n",
      "Merged NA19391\n",
      "Merged NA19468\n",
      "Merged HG00438\n",
      "Merged HG00558\n",
      "Merged HG00639\n",
      "Merged HG00658\n",
      "Merged HG00673\n",
      "Merged HG01071\n",
      "Merged HG01074\n",
      "Merged HG01081\n",
      "Merged HG01123\n",
      "Merged HG01150\n",
      "Merged HG01252\n",
      "Merged HG01358\n",
      "Merged HG01361\n",
      "Merged HG01928\n",
      "Merged HG01934\n",
      "Merged HG01940\n",
      "Merged HG01960\n",
      "Merged HG01969\n",
      "Merged HG01978\n",
      "Merged HG02071\n",
      "Merged HG02074\n",
      "Merged HG02135\n",
      "Merged HG02257\n",
      "Merged HG02668\n",
      "Merged HG02717\n",
      "Merged HG02809\n",
      "Merged HG03017\n",
      "Merged HG03041\n",
      "Merged HG03239\n",
      "Merged HG03516\n",
      "Merged HG03804\n",
      "Merged HG03942\n",
      "Merged HG04157\n",
      "Merged HG04160\n",
      "Merged HG04204\n",
      "Merged HG04228\n",
      "Merged HG00423\n",
      "Merged HG00544\n",
      "Merged HG01943\n",
      "Merged HG02735\n",
      "Merged HG03225\n",
      "Merged HG00126\n",
      "Merged HG00128\n",
      "Merged HG00133\n",
      "Merged HG00146\n",
      "Merged HG00232\n",
      "Merged HG00290\n",
      "Merged HG00320\n",
      "Merged HG00321\n",
      "Merged HG02178\n",
      "Merged HG02583\n",
      "Merged HG03270\n",
      "Merged HG03583\n",
      "Merged HG03874\n",
      "Merged NA18505\n",
      "Merged NA18508\n",
      "Merged NA18608\n",
      "Merged NA18620\n",
      "Merged NA18952\n",
      "Merged NA18974\n",
      "Merged NA18976\n",
      "Merged NA19036\n",
      "Merged NA19443\n",
      "Merged NA19700\n",
      "Merged NA20870\n",
      "Merged NA21093\n",
      "Merged NA21106\n",
      "Merged NA21110\n",
      "Merged HG00735\n",
      "Merged HG02155\n",
      "Merged HG02630\n",
      "Merged HG03139\n",
      "Merged HG03470\n",
      "Merged HG03579\n",
      "Merged NA20805\n",
      "Merged HG01109\n",
      "Merged HG02055\n",
      "Merged HG02080\n",
      "Merged HG02109\n",
      "Merged HG02723\n",
      "Merged HG02818\n",
      "Merged HG03098\n",
      "Merged HG03486\n",
      "Merged NA18906\n",
      "Merged NA20129\n",
      "Merged NA21309\n",
      "Merged HG00097\n",
      "Merged HG00235\n",
      "Merged HG00253\n",
      "Merged HG00272\n",
      "Merged HG00329\n",
      "Merged HG00344\n",
      "Merged HG00350\n",
      "Merged HG01167\n",
      "Merged HG03369\n",
      "Merged HG03521\n",
      "Merged HG03784\n",
      "Merged NA18565\n",
      "Merged NA19682\n",
      "Merged NA19776\n",
      "Merged NA19835\n",
      "Merged NA19909\n",
      "Merged NA20282\n",
      "Merged NA20346\n",
      "Merged NA20809\n",
      "Merged NA20850\n",
      "Merged NA21102\n",
      "Merged NA21144\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>asm_hap1</th>\n",
       "      <th>asm_hap2</th>\n",
       "      <th>cenSatAnnotations_hap1</th>\n",
       "      <th>cenSatAnnotations_hap2</th>\n",
       "      <th>censat_diploid_bed</th>\n",
       "      <th>additional_annotations_array</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG00408</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00408_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG00597</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG00597_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG01192</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG01192_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG01192_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG01261</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG01261_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG01261_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG02015</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG02015_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG02015_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  sample_id                                           asm_hap1  \\\n",
       "0   HG00408  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "1   HG00597  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "2   HG01192  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "3   HG01261  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "4   HG02015  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "\n",
       "                                            asm_hap2  \\\n",
       "0  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "1  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "2  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "3  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "4  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "\n",
       "                              cenSatAnnotations_hap1  \\\n",
       "0  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "1  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "2  /private/groups/hprc/qc/batch1/censat/HG01192_...   \n",
       "3  /private/groups/hprc/qc/batch1/censat/HG01261_...   \n",
       "4  /private/groups/hprc/qc/batch1/censat/HG02015_...   \n",
       "\n",
       "                              cenSatAnnotations_hap2  \\\n",
       "0  /private/groups/hprc/qc/batch1/censat/HG00408_...   \n",
       "1  /private/groups/hprc/qc/batch1/censat/HG00597_...   \n",
       "2  /private/groups/hprc/qc/batch1/censat/HG01192_...   \n",
       "3  /private/groups/hprc/qc/batch1/censat/HG01261_...   \n",
       "4  /private/groups/hprc/qc/batch1/censat/HG02015_...   \n",
       "\n",
       "                                  censat_diploid_bed  \\\n",
       "0  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "1  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "2  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "3  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "4  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "\n",
       "                        additional_annotations_array  \n",
       "0  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...  \n",
       "1  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...  \n",
       "2  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...  \n",
       "3  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...  \n",
       "4  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...  "
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# add a column for saving diploid censat bed files\n",
    "censat_table_diploid[\"censat_diploid_bed\"] = \"\"\n",
    "\n",
    "# make a directory for saving diploid censat bed files\n",
    "diploid_censat_dir = f'{os.getcwd()}/diploid_censat_beds'\n",
    "!mkdir -p {diploid_censat_dir}\n",
    "\n",
    "censat_diploid_list = []\n",
    "additional_annotations_array_list = []\n",
    "# iterate over rows\n",
    "for i in range(len(censat_table_diploid)):\n",
    "    sample = censat_table_diploid[\"sample_id\"][i]\n",
    "    censat_bed_hap1 = censat_table_diploid[\"cenSatAnnotations_hap1\"][i]\n",
    "    censat_bed_hap2 = censat_table_diploid[\"cenSatAnnotations_hap2\"][i]\n",
    "    \n",
    "    # concat hap1 and hap2 censat bed files into a single bed file\n",
    "    #!cat {censat_bed_hap1} {censat_bed_hap2} | bedtools sort -i - > {diploid_censat_dir}/{sample}_dip_hprc_r2_v1.cenSat.bed\n",
    "    #!cat {censat_bed_hap1} {censat_bed_hap2} | bedtools sort -i - | grep -i \"rDNA\" | awk '{{print $$1\"\\t\"$$2\"\\t\"$$3}}' > {diploid_censat_dir}/{sample}_dip_hprc_r2_v1.cenSat.rDNA.bed\n",
    "    \n",
    "    print(f'Merged {sample}')\n",
    "    # add new bed to the table\n",
    "    censat_diploid_list.append(f'{diploid_censat_dir}/{sample}_dip_hprc_r2_v1.cenSat.bed')\n",
    "    additional_annotations_array_list.append([f'{diploid_censat_dir}/{sample}_dip_hprc_r2_v1.cenSat.rDNA.bed'])\n",
    "\n",
    "censat_table_diploid[\"censat_diploid_bed\"] = censat_diploid_list\n",
    "censat_table_diploid[\"additional_annotations_array\"] = additional_annotations_array_list\n",
    "censat_table_diploid.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/private/groups/hprc/qc_hmm_flagger/hprc_intermediate_assembly/assembly_qc/batch1/hmm_flagger/diploid_censat_beds/HG00408_dip_hprc_r2_v1.cenSat.bed',\n",
       " '/private/groups/hprc/qc_hmm_flagger/hprc_intermediate_assembly/assembly_qc/batch1/hmm_flagger/diploid_censat_beds/HG00597_dip_hprc_r2_v1.cenSat.bed',\n",
       " '/private/groups/hprc/qc_hmm_flagger/hprc_intermediate_assembly/assembly_qc/batch1/hmm_flagger/diploid_censat_beds/HG01192_dip_hprc_r2_v1.cenSat.bed',\n",
       " '/private/groups/hprc/qc_hmm_flagger/hprc_intermediate_assembly/assembly_qc/batch1/hmm_flagger/diploid_censat_beds/HG01261_dip_hprc_r2_v1.cenSat.bed',\n",
       " '/private/groups/hprc/qc_hmm_flagger/hprc_intermediate_assembly/assembly_qc/batch1/hmm_flagger/diploid_censat_beds/HG02015_dip_hprc_r2_v1.cenSat.bed']"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(censat_table_diploid[\"censat_diploid_bed\"])[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[['/private/groups/hprc/qc_hmm_flagger/hprc_intermediate_assembly/assembly_qc/batch1/hmm_flagger/diploid_censat_beds/HG00408_dip_hprc_r2_v1.cenSat.rDNA.bed'],\n",
       " ['/private/groups/hprc/qc_hmm_flagger/hprc_intermediate_assembly/assembly_qc/batch1/hmm_flagger/diploid_censat_beds/HG00597_dip_hprc_r2_v1.cenSat.rDNA.bed'],\n",
       " ['/private/groups/hprc/qc_hmm_flagger/hprc_intermediate_assembly/assembly_qc/batch1/hmm_flagger/diploid_censat_beds/HG01192_dip_hprc_r2_v1.cenSat.rDNA.bed'],\n",
       " ['/private/groups/hprc/qc_hmm_flagger/hprc_intermediate_assembly/assembly_qc/batch1/hmm_flagger/diploid_censat_beds/HG01261_dip_hprc_r2_v1.cenSat.rDNA.bed'],\n",
       " ['/private/groups/hprc/qc_hmm_flagger/hprc_intermediate_assembly/assembly_qc/batch1/hmm_flagger/diploid_censat_beds/HG02015_dip_hprc_r2_v1.cenSat.rDNA.bed']]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(censat_table_diploid[\"additional_annotations_array\"])[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HG00408#1#CM085953.1\t132949629\t132952668\r\n",
      "HG00408#1#CM085961.1\t34643331\t34646412\r\n",
      "HG00408#1#CM085964.1\t29763394\t29766442\r\n",
      "HG00408#1#CM085964.1\t30693954\t30697014\r\n",
      "HG00408#1#JBHDVK010000001.1\t19274\t112653\r\n",
      "HG00408#1#JBHDVK010000001.1\t2651597\t2654675\r\n",
      "HG00408#1#JBHDVK010000008.1\t6440370\t6851003\r\n",
      "HG00408#1#JBHDVK010000008.1\t7045506\t7660493\r\n",
      "HG00408#1#JBHDVK010000008.1\t12266867\t12268108\r\n",
      "HG00408#1#JBHDVK010000008.1\t13361951\t13365007\r\n"
     ]
    }
   ],
   "source": [
    "!cat /private/groups/hprc/qc_hmm_flagger/hprc_intermediate_assembly/assembly_qc/batch1/hmm_flagger/diploid_censat_beds/HG00408_dip_hprc_r2_v1.cenSat.rDNA.bed | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/private/groups/hprc/qc_hmm_flagger/hprc_intermediate_assembly/assembly_qc/batch1/hmm_flagger'"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.getcwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read ont and hifi reads table\n",
    "hifi_reads_table = pd.read_csv('read_tables/hifi_reads_table.csv')\n",
    "ont_reads_table = pd.read_csv('read_tables/ont_reads_table.csv')\n",
    "\n",
    "# merge with censat table\n",
    "ont_data_table = pd.merge(censat_table_diploid, ont_reads_table, on='sample_id',  how='inner')\n",
    "hifi_data_table = pd.merge(censat_table_diploid, hifi_reads_table, on='sample_id',  how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of rows: 47\n"
     ]
    }
   ],
   "source": [
    "hprc_y1_table = pd.read_csv('hprc_year1_sample_metadata.txt', sep=\"\\t\")\n",
    "print(\"Number of rows:\", len(hprc_y1_table))\n",
    "y1_samples = list(hprc_y1_table[\"Sample\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['HG01123',\n",
       " 'HG01258',\n",
       " 'HG01358',\n",
       " 'HG01361',\n",
       " 'HG01891',\n",
       " 'HG02257',\n",
       " 'HG02486',\n",
       " 'HG02559',\n",
       " 'HG02572',\n",
       " 'HG03516',\n",
       " 'HG00438',\n",
       " 'HG00621',\n",
       " 'HG00673',\n",
       " 'HG00735',\n",
       " 'HG00741',\n",
       " 'HG01071',\n",
       " 'HG01106',\n",
       " 'HG01175',\n",
       " 'HG01928',\n",
       " 'HG01952',\n",
       " 'HG01978',\n",
       " 'HG02148',\n",
       " 'HG02622',\n",
       " 'HG02630',\n",
       " 'HG02717',\n",
       " 'HG02886',\n",
       " 'HG03453',\n",
       " 'HG03540',\n",
       " 'HG03579',\n",
       " 'HG002',\n",
       " 'HG005',\n",
       " 'HG00733',\n",
       " 'HG01109',\n",
       " 'HG01243',\n",
       " 'HG02080',\n",
       " 'HG02109',\n",
       " 'HG02145',\n",
       " 'HG02723',\n",
       " 'HG02818',\n",
       " 'HG03486',\n",
       " 'HG03492',\n",
       " 'NA18906',\n",
       " 'NA19240',\n",
       " 'NA20129',\n",
       " 'NA21309',\n",
       " 'HG02055',\n",
       " 'HG03098']"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "y1_samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "ont_data_table[\"is_year1\"] = ont_data_table[\"sample_id\"].apply(lambda x: \"true\" if x in y1_samples else \"false\")\n",
    "hifi_data_table[\"is_year1\"] = hifi_data_table[\"sample_id\"].apply(lambda x: \"true\" if x in y1_samples else \"false\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "ont_data_table = ont_data_table.sort_values(by=[\"is_year1\", \"sample_id\"], ascending=False)\n",
    "hifi_data_table = hifi_data_table.sort_values(by=[\"is_year1\", \"sample_id\"], ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of rows for final HiFi table: 141\n",
      "Number of rows for final HiFi table (Year1): 40\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>asm_hap1</th>\n",
       "      <th>asm_hap2</th>\n",
       "      <th>cenSatAnnotations_hap1</th>\n",
       "      <th>cenSatAnnotations_hap2</th>\n",
       "      <th>censat_diploid_bed</th>\n",
       "      <th>additional_annotations_array</th>\n",
       "      <th>read_files</th>\n",
       "      <th>number_of_read_files</th>\n",
       "      <th>total_coverage</th>\n",
       "      <th>coverage</th>\n",
       "      <th>mapper_preset</th>\n",
       "      <th>kmer_size</th>\n",
       "      <th>number_of_cores_per_task</th>\n",
       "      <th>hmm_flagger_window_size</th>\n",
       "      <th>hmm_flagger_alpha_tsv</th>\n",
       "      <th>is_year1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>NA21309</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/NA21309_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/NA21309_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/submissions/3A25CF8A-...</td>\n",
       "      <td>4</td>\n",
       "      <td>45.34</td>\n",
       "      <td>[11.84, 13.56, 13.32, 6.62]</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>16</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "      <td>true</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>NA20129</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/NA20129_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/NA20129_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/submissions/3A25CF8A-...</td>\n",
       "      <td>5</td>\n",
       "      <td>42.33</td>\n",
       "      <td>[10.52, 11.08, 11.16, 3.26, 6.31]</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>12</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "      <td>true</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>NA18906</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/NA18906_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/NA18906_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/submissions/3A25CF8A-...</td>\n",
       "      <td>4</td>\n",
       "      <td>51.45</td>\n",
       "      <td>[12.68, 13.23, 12.52, 13.02]</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>16</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "      <td>true</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>HG03579</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG03579_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG03579_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/submissions/3A25CF8A-...</td>\n",
       "      <td>6</td>\n",
       "      <td>84.47</td>\n",
       "      <td>[12.94, 14.54, 14.68, 14.19, 16.84, 11.28]</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>10</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "      <td>true</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>HG03540</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG03540_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG03540_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/submissions/3A25CF8A-...</td>\n",
       "      <td>4</td>\n",
       "      <td>58.53</td>\n",
       "      <td>[15.01, 14.59, 14.66, 14.27]</td>\n",
       "      <td>lr:hqae</td>\n",
       "      <td>25</td>\n",
       "      <td>16</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "      <td>true</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    sample_id                                           asm_hap1  \\\n",
       "140   NA21309  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "139   NA20129  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "138   NA18906  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "128   HG03579  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "55    HG03540  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "\n",
       "                                              asm_hap2  \\\n",
       "140  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "139  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "138  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "128  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "55   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "\n",
       "                                cenSatAnnotations_hap1  \\\n",
       "140  /private/groups/hprc/qc/batch1/censat/NA21309_...   \n",
       "139  /private/groups/hprc/qc/batch1/censat/NA20129_...   \n",
       "138  /private/groups/hprc/qc/batch1/censat/NA18906_...   \n",
       "128  /private/groups/hprc/qc/batch1/censat/HG03579_...   \n",
       "55   /private/groups/hprc/qc/batch1/censat/HG03540_...   \n",
       "\n",
       "                                cenSatAnnotations_hap2  \\\n",
       "140  /private/groups/hprc/qc/batch1/censat/NA21309_...   \n",
       "139  /private/groups/hprc/qc/batch1/censat/NA20129_...   \n",
       "138  /private/groups/hprc/qc/batch1/censat/NA18906_...   \n",
       "128  /private/groups/hprc/qc/batch1/censat/HG03579_...   \n",
       "55   /private/groups/hprc/qc/batch1/censat/HG03540_...   \n",
       "\n",
       "                                    censat_diploid_bed  \\\n",
       "140  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "139  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "138  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "128  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "55   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "\n",
       "                          additional_annotations_array  \\\n",
       "140  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "139  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "138  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "128  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "55   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "\n",
       "                                            read_files  number_of_read_files  \\\n",
       "140  ['s3://human-pangenomics/submissions/3A25CF8A-...                     4   \n",
       "139  ['s3://human-pangenomics/submissions/3A25CF8A-...                     5   \n",
       "138  ['s3://human-pangenomics/submissions/3A25CF8A-...                     4   \n",
       "128  ['s3://human-pangenomics/submissions/3A25CF8A-...                     6   \n",
       "55   ['s3://human-pangenomics/submissions/3A25CF8A-...                     4   \n",
       "\n",
       "     total_coverage                                    coverage mapper_preset  \\\n",
       "140           45.34                 [11.84, 13.56, 13.32, 6.62]       lr:hqae   \n",
       "139           42.33           [10.52, 11.08, 11.16, 3.26, 6.31]       lr:hqae   \n",
       "138           51.45                [12.68, 13.23, 12.52, 13.02]       lr:hqae   \n",
       "128           84.47  [12.94, 14.54, 14.68, 14.19, 16.84, 11.28]       lr:hqae   \n",
       "55            58.53                [15.01, 14.59, 14.66, 14.27]       lr:hqae   \n",
       "\n",
       "     kmer_size  number_of_cores_per_task  hmm_flagger_window_size  \\\n",
       "140         25                        16                    16000   \n",
       "139         25                        12                    16000   \n",
       "138         25                        16                    16000   \n",
       "128         25                        10                    16000   \n",
       "55          25                        16                    16000   \n",
       "\n",
       "                                 hmm_flagger_alpha_tsv is_year1  \n",
       "140  https://raw.githubusercontent.com/mobinasri/fl...     true  \n",
       "139  https://raw.githubusercontent.com/mobinasri/fl...     true  \n",
       "138  https://raw.githubusercontent.com/mobinasri/fl...     true  \n",
       "128  https://raw.githubusercontent.com/mobinasri/fl...     true  \n",
       "55   https://raw.githubusercontent.com/mobinasri/fl...     true  "
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(f\"Number of rows for final HiFi table: {len(hifi_data_table)}\")\n",
    "print(f\"Number of rows for final HiFi table (Year1): {sum(hifi_data_table['is_year1'] == 'true')}\")\n",
    "hifi_data_table.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of rows for final ONT table: 170\n",
      "Number of rows for final ONT table (Year1): 29\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>asm_hap1</th>\n",
       "      <th>asm_hap2</th>\n",
       "      <th>cenSatAnnotations_hap1</th>\n",
       "      <th>cenSatAnnotations_hap2</th>\n",
       "      <th>censat_diploid_bed</th>\n",
       "      <th>additional_annotations_array</th>\n",
       "      <th>read_files_downsampled</th>\n",
       "      <th>number_of_read_files_downsampled</th>\n",
       "      <th>total_coverage_downsampled</th>\n",
       "      <th>...</th>\n",
       "      <th>kmer_size</th>\n",
       "      <th>read_files</th>\n",
       "      <th>number_of_read_files</th>\n",
       "      <th>total_coverage</th>\n",
       "      <th>coverage</th>\n",
       "      <th>number_of_cores_per_task</th>\n",
       "      <th>sequencing_chemistry</th>\n",
       "      <th>hmm_flagger_window_size</th>\n",
       "      <th>hmm_flagger_alpha_tsv</th>\n",
       "      <th>is_year1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>HG03579</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG03579_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG03579_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG03579/...</td>\n",
       "      <td>4</td>\n",
       "      <td>61.24</td>\n",
       "      <td>...</td>\n",
       "      <td>15</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG03579/...</td>\n",
       "      <td>9</td>\n",
       "      <td>95.33</td>\n",
       "      <td>[5.51, 4.3, 2.78, 10.72, 12.87, 10.78, 16.91, ...</td>\n",
       "      <td>7</td>\n",
       "      <td>R941</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "      <td>true</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>HG03540</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG03540_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG03540_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG03540/...</td>\n",
       "      <td>4</td>\n",
       "      <td>66.73</td>\n",
       "      <td>...</td>\n",
       "      <td>15</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG03540/...</td>\n",
       "      <td>5</td>\n",
       "      <td>69.44</td>\n",
       "      <td>[19.56, 28.57, 6.77, 11.83, 2.71]</td>\n",
       "      <td>12</td>\n",
       "      <td>R941</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "      <td>true</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>HG03516</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG03516_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG03516_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG03516/...</td>\n",
       "      <td>4</td>\n",
       "      <td>65.43</td>\n",
       "      <td>...</td>\n",
       "      <td>15</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG03516/...</td>\n",
       "      <td>9</td>\n",
       "      <td>126.17</td>\n",
       "      <td>[10.45, 12.86, 13.36, 12.69, 16.85, 13.58, 17....</td>\n",
       "      <td>7</td>\n",
       "      <td>R941</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "      <td>true</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>HG03453</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG03453_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG03453_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG03453/...</td>\n",
       "      <td>4</td>\n",
       "      <td>60.64</td>\n",
       "      <td>...</td>\n",
       "      <td>15</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG03453/...</td>\n",
       "      <td>5</td>\n",
       "      <td>61.51</td>\n",
       "      <td>[30.59, 22.17, 3.24, 4.64, 0.87]</td>\n",
       "      <td>12</td>\n",
       "      <td>R941</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "      <td>true</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>HG02886</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>s3://human-pangenomics/submissions/DC27718F-5F...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG02886_...</td>\n",
       "      <td>/private/groups/hprc/qc/batch1/censat/HG02886_...</td>\n",
       "      <td>/private/groups/hprc/qc_hmm_flagger/hprc_inter...</td>\n",
       "      <td>[/private/groups/hprc/qc_hmm_flagger/hprc_inte...</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02886/...</td>\n",
       "      <td>5</td>\n",
       "      <td>64.07</td>\n",
       "      <td>...</td>\n",
       "      <td>15</td>\n",
       "      <td>['s3://human-pangenomics/working/HPRC/HG02886/...</td>\n",
       "      <td>5</td>\n",
       "      <td>64.07</td>\n",
       "      <td>[20.24, 16.83, 9.84, 10.72, 6.44]</td>\n",
       "      <td>12</td>\n",
       "      <td>R941</td>\n",
       "      <td>16000</td>\n",
       "      <td>https://raw.githubusercontent.com/mobinasri/fl...</td>\n",
       "      <td>true</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    sample_id                                           asm_hap1  \\\n",
       "147   HG03579  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "55    HG03540  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "103   HG03516  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "54    HG03453  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "53    HG02886  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "\n",
       "                                              asm_hap2  \\\n",
       "147  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "55   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "103  s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "54   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "53   s3://human-pangenomics/submissions/DC27718F-5F...   \n",
       "\n",
       "                                cenSatAnnotations_hap1  \\\n",
       "147  /private/groups/hprc/qc/batch1/censat/HG03579_...   \n",
       "55   /private/groups/hprc/qc/batch1/censat/HG03540_...   \n",
       "103  /private/groups/hprc/qc/batch1/censat/HG03516_...   \n",
       "54   /private/groups/hprc/qc/batch1/censat/HG03453_...   \n",
       "53   /private/groups/hprc/qc/batch1/censat/HG02886_...   \n",
       "\n",
       "                                cenSatAnnotations_hap2  \\\n",
       "147  /private/groups/hprc/qc/batch1/censat/HG03579_...   \n",
       "55   /private/groups/hprc/qc/batch1/censat/HG03540_...   \n",
       "103  /private/groups/hprc/qc/batch1/censat/HG03516_...   \n",
       "54   /private/groups/hprc/qc/batch1/censat/HG03453_...   \n",
       "53   /private/groups/hprc/qc/batch1/censat/HG02886_...   \n",
       "\n",
       "                                    censat_diploid_bed  \\\n",
       "147  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "55   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "103  /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "54   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "53   /private/groups/hprc/qc_hmm_flagger/hprc_inter...   \n",
       "\n",
       "                          additional_annotations_array  \\\n",
       "147  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "55   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "103  [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "54   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "53   [/private/groups/hprc/qc_hmm_flagger/hprc_inte...   \n",
       "\n",
       "                                read_files_downsampled  \\\n",
       "147  ['s3://human-pangenomics/working/HPRC/HG03579/...   \n",
       "55   ['s3://human-pangenomics/working/HPRC/HG03540/...   \n",
       "103  ['s3://human-pangenomics/working/HPRC/HG03516/...   \n",
       "54   ['s3://human-pangenomics/working/HPRC/HG03453/...   \n",
       "53   ['s3://human-pangenomics/working/HPRC/HG02886/...   \n",
       "\n",
       "     number_of_read_files_downsampled  total_coverage_downsampled    ...     \\\n",
       "147                                 4                       61.24    ...      \n",
       "55                                  4                       66.73    ...      \n",
       "103                                 4                       65.43    ...      \n",
       "54                                  4                       60.64    ...      \n",
       "53                                  5                       64.07    ...      \n",
       "\n",
       "     kmer_size                                         read_files  \\\n",
       "147         15  ['s3://human-pangenomics/working/HPRC/HG03579/...   \n",
       "55          15  ['s3://human-pangenomics/working/HPRC/HG03540/...   \n",
       "103         15  ['s3://human-pangenomics/working/HPRC/HG03516/...   \n",
       "54          15  ['s3://human-pangenomics/working/HPRC/HG03453/...   \n",
       "53          15  ['s3://human-pangenomics/working/HPRC/HG02886/...   \n",
       "\n",
       "     number_of_read_files total_coverage  \\\n",
       "147                     9          95.33   \n",
       "55                      5          69.44   \n",
       "103                     9         126.17   \n",
       "54                      5          61.51   \n",
       "53                      5          64.07   \n",
       "\n",
       "                                              coverage  \\\n",
       "147  [5.51, 4.3, 2.78, 10.72, 12.87, 10.78, 16.91, ...   \n",
       "55                   [19.56, 28.57, 6.77, 11.83, 2.71]   \n",
       "103  [10.45, 12.86, 13.36, 12.69, 16.85, 13.58, 17....   \n",
       "54                    [30.59, 22.17, 3.24, 4.64, 0.87]   \n",
       "53                   [20.24, 16.83, 9.84, 10.72, 6.44]   \n",
       "\n",
       "     number_of_cores_per_task sequencing_chemistry  hmm_flagger_window_size  \\\n",
       "147                         7                 R941                    16000   \n",
       "55                         12                 R941                    16000   \n",
       "103                         7                 R941                    16000   \n",
       "54                         12                 R941                    16000   \n",
       "53                         12                 R941                    16000   \n",
       "\n",
       "                                 hmm_flagger_alpha_tsv  is_year1  \n",
       "147  https://raw.githubusercontent.com/mobinasri/fl...      true  \n",
       "55   https://raw.githubusercontent.com/mobinasri/fl...      true  \n",
       "103  https://raw.githubusercontent.com/mobinasri/fl...      true  \n",
       "54   https://raw.githubusercontent.com/mobinasri/fl...      true  \n",
       "53   https://raw.githubusercontent.com/mobinasri/fl...      true  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(f\"Number of rows for final ONT table: {len(ont_data_table)}\")\n",
    "print(f\"Number of rows for final ONT table (Year1): {sum(ont_data_table['is_year1'] == 'true')}\")\n",
    "ont_data_table.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "os.makedirs(\"ont\", exist_ok=True)\n",
    "ont_data_table.to_csv('ont/hmm_flagger_ont_data_table.csv', index=False)\n",
    "\n",
    "os.makedirs(\"hifi\", exist_ok=True)\n",
    "hifi_data_table.to_csv('hifi/hmm_flagger_hifi_data_table.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "colab": {
   "provenance": []
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
